{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "auth_params = {\"email\":\"francolsimonetti@gmail.com\",\"password\":\"pAgaMFf6egSAW5T\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# api_host = \"http://localhost:5050/api\"\n",
    "\n",
    "# api_key = None\n",
    "# s = requests.Session()\n",
    "# try:\n",
    "#     r = s.post(api_host+'/auth/', data=auth_params)\n",
    "#     if(r.status_code == 200):\n",
    "#         #Lets store the api key in a new variable and use it again in new requests\n",
    "#         json_response = r.json()\n",
    "#         api_key = json_response.get(\"token\")\n",
    "#         print(api_key + \"This is your user API key.\") #Comment this line if you don't want your API key to show up in the terminal\n",
    "#     else:\n",
    "#         print(r.status_code)\n",
    "#         print(r.text)\n",
    "# except requests.exceptions.RequestException as req_ex:\n",
    "#     print(req_ex)\n",
    "#     print(\"Something went wrong with the request.\")\n",
    "\n",
    "# if api_key:\n",
    "#     #Add the api key to the requests headers of the requests Session object in order to use the restricted endpoints.\n",
    "#     s.headers.update({\"Authorization\": \"Bearer %s\" % api_key}) \n",
    "#     #Lets get all the diseases associated to a gene eg. APP (EntrezID 351) and restricted by a source.\n",
    "#     gda_response = s.get(api_host+'/gda/gene/351', params={'source':'UNIPROT'})\n",
    "#     print(gda_response.json())\n",
    "\n",
    "# if s:\n",
    "#     s.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# este dataset viene de la pagina de downloads (https://www.disgenet.org/downloads; Octubre 2020)\n",
    "vda = pd.read_csv('../../data/curated_variant_disease_associations.tsv.gz', sep='\\t')\n",
    "vda.columns = vda.columns.str.lower().str.replace(' ',\"_\").str.replace(\"-\",'_').str.replace('/','_')\n",
    "vda = vda.rename(columns={'snpid':'snp_id'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snps unicos en el dataset: 168051 vs (261227, 16)\n"
     ]
    }
   ],
   "source": [
    "print(f'snps unicos en el dataset: {len(vda.snp_id.unique())} vs {vda.shape}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Mappings UniProts: uniprots id con entrez gene id (https://www.disgenet.org/downloads, Mappings)\n",
    "uniprots = pd.read_csv('../../data/mapa_geneid_4_uniprot_crossref.tsv.gz', sep='\\t', compression='gzip')\n",
    "uniprots = uniprots.rename(columns={'UniProtKB':'uniprot_acc', 'GENEID':'gene_id'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The file contains the mappings of DisGeNET variants (dbSNP Identifiers)\n",
    "# to NCBI Entrez identifiers according to dbSNP database (https://www.disgenet.org/downloads, Variant-Gene Mappings File)\n",
    "variant_gene = pd.read_csv('variant_to_gene_mappings.tsv.gz', sep='\\t', compression='gzip')\n",
    "variant_gene = variant_gene.rename(columns={'snpId':'snp_id', 'geneId':'gene_id', 'geneSymbol': 'gene_name', 'sourceId':'source_id'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot_acc</th>\n",
       "      <th>gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P04217</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P11245</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P00813</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P19022</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Q9Y243</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  uniprot_acc  gene_id\n",
       "0      P04217        1\n",
       "1      P11245       10\n",
       "2      P00813      100\n",
       "3      P19022     1000\n",
       "4      Q9Y243    10000"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprots.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snp_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>source_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs7980288</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs7980288</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs3832852</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs669</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs669</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      snp_id  gene_id gene_name source_id\n",
       "0  rs7980288        2       A2M       VEP\n",
       "1  rs7980288        2       A2M     DBSNP\n",
       "2  rs3832852        2       A2M     DBSNP\n",
       "3      rs669        2       A2M       VEP\n",
       "4      rs669        2       A2M     DBSNP"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variant_gene.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snp_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>source_id</th>\n",
       "      <th>uniprot_acc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs7980288</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>VEP</td>\n",
       "      <td>P01023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs7980288</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>P01023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs3832852</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>P01023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs669</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>VEP</td>\n",
       "      <td>P01023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs669</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>P01023</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      snp_id  gene_id gene_name source_id uniprot_acc\n",
       "0  rs7980288        2       A2M       VEP      P01023\n",
       "1  rs7980288        2       A2M     DBSNP      P01023\n",
       "2  rs3832852        2       A2M     DBSNP      P01023\n",
       "3      rs669        2       A2M       VEP      P01023\n",
       "4      rs669        2       A2M     DBSNP      P01023"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# agrego los uniprots\n",
    "variant_gene = variant_gene.merge(uniprots)\n",
    "variant_gene.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snp_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>dsi</th>\n",
       "      <th>dpi</th>\n",
       "      <th>diseaseid</th>\n",
       "      <th>diseasename</th>\n",
       "      <th>diseasetype</th>\n",
       "      <th>diseaseclass</th>\n",
       "      <th>diseasesemantictype</th>\n",
       "      <th>score</th>\n",
       "      <th>ei</th>\n",
       "      <th>yearinitial</th>\n",
       "      <th>yearfinal</th>\n",
       "      <th>nofpmids</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000005</td>\n",
       "      <td>21</td>\n",
       "      <td>33060745</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C0200638</td>\n",
       "      <td>Eosinophil count procedure</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Laboratory Procedure</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.04</td>\n",
       "      <td>C0023467</td>\n",
       "      <td>Leukemia, Myelocytic, Acute</td>\n",
       "      <td>disease</td>\n",
       "      <td>C04</td>\n",
       "      <td>Neoplastic Process</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>C1858517</td>\n",
       "      <td>SPINAL MUSCULAR ATROPHY WITH RESPIRATORY DISTR...</td>\n",
       "      <td>disease</td>\n",
       "      <td>C16;C08;C10</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>1</td>\n",
       "      <td>CLINVAR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000096</td>\n",
       "      <td>4</td>\n",
       "      <td>38691214</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C1305855</td>\n",
       "      <td>Body mass index</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Clinical Attribute</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2019.0</td>\n",
       "      <td>2019.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs10001106</td>\n",
       "      <td>4</td>\n",
       "      <td>10125817</td>\n",
       "      <td>0.925</td>\n",
       "      <td>0.12</td>\n",
       "      <td>C0003868</td>\n",
       "      <td>Arthritis, Gouty</td>\n",
       "      <td>disease</td>\n",
       "      <td>C16;C18;C05</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2010.0</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>3</td>\n",
       "      <td>GWASDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261222</th>\n",
       "      <td>rs9999118</td>\n",
       "      <td>4</td>\n",
       "      <td>123130312</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.04</td>\n",
       "      <td>C0022104</td>\n",
       "      <td>Irritable Bowel Syndrome</td>\n",
       "      <td>disease</td>\n",
       "      <td>C06</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2014.0</td>\n",
       "      <td>2014.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT;GWASDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261223</th>\n",
       "      <td>rs999921351</td>\n",
       "      <td>8</td>\n",
       "      <td>86628939</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>C1849792</td>\n",
       "      <td>Achromatopsia 3</td>\n",
       "      <td>disease</td>\n",
       "      <td>C23;C11;C10</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>CLINVAR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261224</th>\n",
       "      <td>rs999943</td>\n",
       "      <td>6</td>\n",
       "      <td>33656956</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>C0028754</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>disease</td>\n",
       "      <td>C23;C18</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT;GWASDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261225</th>\n",
       "      <td>rs999944</td>\n",
       "      <td>2</td>\n",
       "      <td>64822719</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>C0520679</td>\n",
       "      <td>Sleep Apnea, Obstructive</td>\n",
       "      <td>disease</td>\n",
       "      <td>C08;C10</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261226</th>\n",
       "      <td>rs9999820</td>\n",
       "      <td>4</td>\n",
       "      <td>155197173</td>\n",
       "      <td>0.925</td>\n",
       "      <td>0.08</td>\n",
       "      <td>C0085215</td>\n",
       "      <td>Ovarian Failure, Premature</td>\n",
       "      <td>disease</td>\n",
       "      <td>C13;C19</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>261227 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              snp_id chromosome   position    dsi   dpi diseaseid  \\\n",
       "0          rs1000005         21   33060745    NaN   NaN  C0200638   \n",
       "1         rs10000770          4  142693109  1.000  0.04  C0023467   \n",
       "2       rs1000091588         11   68935374  1.000  0.12  C1858517   \n",
       "3          rs1000096          4   38691214    NaN   NaN  C1305855   \n",
       "4         rs10001106          4   10125817  0.925  0.12  C0003868   \n",
       "...              ...        ...        ...    ...   ...       ...   \n",
       "261222     rs9999118          4  123130312  1.000  0.04  C0022104   \n",
       "261223   rs999921351          8   86628939  1.000  0.12  C1849792   \n",
       "261224      rs999943          6   33656956  1.000  0.08  C0028754   \n",
       "261225      rs999944          2   64822719  1.000  0.08  C0520679   \n",
       "261226     rs9999820          4  155197173  0.925  0.08  C0085215   \n",
       "\n",
       "                                              diseasename diseasetype  \\\n",
       "0                              Eosinophil count procedure   phenotype   \n",
       "1                             Leukemia, Myelocytic, Acute     disease   \n",
       "2       SPINAL MUSCULAR ATROPHY WITH RESPIRATORY DISTR...     disease   \n",
       "3                                         Body mass index   phenotype   \n",
       "4                                        Arthritis, Gouty     disease   \n",
       "...                                                   ...         ...   \n",
       "261222                           Irritable Bowel Syndrome     disease   \n",
       "261223                                    Achromatopsia 3     disease   \n",
       "261224                                            Obesity     disease   \n",
       "261225                           Sleep Apnea, Obstructive     disease   \n",
       "261226                         Ovarian Failure, Premature     disease   \n",
       "\n",
       "       diseaseclass   diseasesemantictype  score   ei  yearinitial  yearfinal  \\\n",
       "0               NaN  Laboratory Procedure    0.7  1.0       2016.0     2016.0   \n",
       "1               C04    Neoplastic Process    0.7  1.0       2017.0     2017.0   \n",
       "2       C16;C08;C10   Disease or Syndrome    0.7  1.0       2007.0     2007.0   \n",
       "3               NaN    Clinical Attribute    0.7  1.0       2019.0     2019.0   \n",
       "4       C16;C18;C05   Disease or Syndrome    0.7  1.0       2010.0     2013.0   \n",
       "...             ...                   ...    ...  ...          ...        ...   \n",
       "261222          C06   Disease or Syndrome    0.8  1.0       2014.0     2014.0   \n",
       "261223  C23;C11;C10   Disease or Syndrome    0.7  1.0       2017.0     2017.0   \n",
       "261224      C23;C18   Disease or Syndrome    0.8  1.0       2009.0     2009.0   \n",
       "261225      C08;C10   Disease or Syndrome    0.7  1.0       2016.0     2016.0   \n",
       "261226      C13;C19   Disease or Syndrome    0.7  1.0       2018.0     2018.0   \n",
       "\n",
       "        nofpmids          source  \n",
       "0              1         GWASCAT  \n",
       "1              1         GWASCAT  \n",
       "2              1         CLINVAR  \n",
       "3              1         GWASCAT  \n",
       "4              3          GWASDB  \n",
       "...          ...             ...  \n",
       "261222         1  GWASCAT;GWASDB  \n",
       "261223         1         CLINVAR  \n",
       "261224         1  GWASCAT;GWASDB  \n",
       "261225         1         GWASCAT  \n",
       "261226         1         GWASCAT  \n",
       "\n",
       "[261227 rows x 16 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vda"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snp_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>dsi</th>\n",
       "      <th>dpi</th>\n",
       "      <th>diseaseid</th>\n",
       "      <th>diseasename</th>\n",
       "      <th>diseasetype</th>\n",
       "      <th>diseaseclass</th>\n",
       "      <th>diseasesemantictype</th>\n",
       "      <th>score</th>\n",
       "      <th>ei</th>\n",
       "      <th>yearinitial</th>\n",
       "      <th>yearfinal</th>\n",
       "      <th>nofpmids</th>\n",
       "      <th>source</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>source_id</th>\n",
       "      <th>uniprot_acc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000005</td>\n",
       "      <td>21</td>\n",
       "      <td>33060745</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C0200638</td>\n",
       "      <td>Eosinophil count procedure</td>\n",
       "      <td>phenotype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Laboratory Procedure</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.04</td>\n",
       "      <td>C0023467</td>\n",
       "      <td>Leukemia, Myelocytic, Acute</td>\n",
       "      <td>disease</td>\n",
       "      <td>C04</td>\n",
       "      <td>Neoplastic Process</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "      <td>8821.0</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>O15327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.04</td>\n",
       "      <td>C0023467</td>\n",
       "      <td>Leukemia, Myelocytic, Acute</td>\n",
       "      <td>disease</td>\n",
       "      <td>C04</td>\n",
       "      <td>Neoplastic Process</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "      <td>8821.0</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>VEP</td>\n",
       "      <td>O15327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>C1858517</td>\n",
       "      <td>SPINAL MUSCULAR ATROPHY WITH RESPIRATORY DISTR...</td>\n",
       "      <td>disease</td>\n",
       "      <td>C16;C08;C10</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>1</td>\n",
       "      <td>CLINVAR</td>\n",
       "      <td>3508.0</td>\n",
       "      <td>IGHMBP2</td>\n",
       "      <td>VEP</td>\n",
       "      <td>P38935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>C1858517</td>\n",
       "      <td>SPINAL MUSCULAR ATROPHY WITH RESPIRATORY DISTR...</td>\n",
       "      <td>disease</td>\n",
       "      <td>C16;C08;C10</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>1</td>\n",
       "      <td>CLINVAR</td>\n",
       "      <td>3508.0</td>\n",
       "      <td>IGHMBP2</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>P38935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489665</th>\n",
       "      <td>rs999921351</td>\n",
       "      <td>8</td>\n",
       "      <td>86628939</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.12</td>\n",
       "      <td>C1849792</td>\n",
       "      <td>Achromatopsia 3</td>\n",
       "      <td>disease</td>\n",
       "      <td>C23;C11;C10</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>1</td>\n",
       "      <td>CLINVAR</td>\n",
       "      <td>54714.0</td>\n",
       "      <td>CNGB3</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Q9NQW8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489666</th>\n",
       "      <td>rs999943</td>\n",
       "      <td>6</td>\n",
       "      <td>33656956</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>C0028754</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>disease</td>\n",
       "      <td>C23;C18</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT;GWASDB</td>\n",
       "      <td>3710.0</td>\n",
       "      <td>ITPR3</td>\n",
       "      <td>VEP</td>\n",
       "      <td>Q14573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489667</th>\n",
       "      <td>rs999943</td>\n",
       "      <td>6</td>\n",
       "      <td>33656956</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>C0028754</td>\n",
       "      <td>Obesity</td>\n",
       "      <td>disease</td>\n",
       "      <td>C23;C18</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT;GWASDB</td>\n",
       "      <td>3710.0</td>\n",
       "      <td>ITPR3</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>Q14573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489668</th>\n",
       "      <td>rs999944</td>\n",
       "      <td>2</td>\n",
       "      <td>64822719</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.08</td>\n",
       "      <td>C0520679</td>\n",
       "      <td>Sleep Apnea, Obstructive</td>\n",
       "      <td>disease</td>\n",
       "      <td>C08;C10</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489669</th>\n",
       "      <td>rs9999820</td>\n",
       "      <td>4</td>\n",
       "      <td>155197173</td>\n",
       "      <td>0.925</td>\n",
       "      <td>0.08</td>\n",
       "      <td>C0085215</td>\n",
       "      <td>Ovarian Failure, Premature</td>\n",
       "      <td>disease</td>\n",
       "      <td>C13;C19</td>\n",
       "      <td>Disease or Syndrome</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>1</td>\n",
       "      <td>GWASCAT</td>\n",
       "      <td>4887.0</td>\n",
       "      <td>NPY2R</td>\n",
       "      <td>DBSNP</td>\n",
       "      <td>P49146</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>489670 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              snp_id chromosome   position    dsi   dpi diseaseid  \\\n",
       "0          rs1000005         21   33060745    NaN   NaN  C0200638   \n",
       "1         rs10000770          4  142693109  1.000  0.04  C0023467   \n",
       "2         rs10000770          4  142693109  1.000  0.04  C0023467   \n",
       "3       rs1000091588         11   68935374  1.000  0.12  C1858517   \n",
       "4       rs1000091588         11   68935374  1.000  0.12  C1858517   \n",
       "...              ...        ...        ...    ...   ...       ...   \n",
       "489665   rs999921351          8   86628939  1.000  0.12  C1849792   \n",
       "489666      rs999943          6   33656956  1.000  0.08  C0028754   \n",
       "489667      rs999943          6   33656956  1.000  0.08  C0028754   \n",
       "489668      rs999944          2   64822719  1.000  0.08  C0520679   \n",
       "489669     rs9999820          4  155197173  0.925  0.08  C0085215   \n",
       "\n",
       "                                              diseasename diseasetype  \\\n",
       "0                              Eosinophil count procedure   phenotype   \n",
       "1                             Leukemia, Myelocytic, Acute     disease   \n",
       "2                             Leukemia, Myelocytic, Acute     disease   \n",
       "3       SPINAL MUSCULAR ATROPHY WITH RESPIRATORY DISTR...     disease   \n",
       "4       SPINAL MUSCULAR ATROPHY WITH RESPIRATORY DISTR...     disease   \n",
       "...                                                   ...         ...   \n",
       "489665                                    Achromatopsia 3     disease   \n",
       "489666                                            Obesity     disease   \n",
       "489667                                            Obesity     disease   \n",
       "489668                           Sleep Apnea, Obstructive     disease   \n",
       "489669                         Ovarian Failure, Premature     disease   \n",
       "\n",
       "       diseaseclass   diseasesemantictype  score   ei  yearinitial  yearfinal  \\\n",
       "0               NaN  Laboratory Procedure    0.7  1.0       2016.0     2016.0   \n",
       "1               C04    Neoplastic Process    0.7  1.0       2017.0     2017.0   \n",
       "2               C04    Neoplastic Process    0.7  1.0       2017.0     2017.0   \n",
       "3       C16;C08;C10   Disease or Syndrome    0.7  1.0       2007.0     2007.0   \n",
       "4       C16;C08;C10   Disease or Syndrome    0.7  1.0       2007.0     2007.0   \n",
       "...             ...                   ...    ...  ...          ...        ...   \n",
       "489665  C23;C11;C10   Disease or Syndrome    0.7  1.0       2017.0     2017.0   \n",
       "489666      C23;C18   Disease or Syndrome    0.8  1.0       2009.0     2009.0   \n",
       "489667      C23;C18   Disease or Syndrome    0.8  1.0       2009.0     2009.0   \n",
       "489668      C08;C10   Disease or Syndrome    0.7  1.0       2016.0     2016.0   \n",
       "489669      C13;C19   Disease or Syndrome    0.7  1.0       2018.0     2018.0   \n",
       "\n",
       "        nofpmids          source  gene_id gene_name source_id uniprot_acc  \n",
       "0              1         GWASCAT      NaN       NaN       NaN         NaN  \n",
       "1              1         GWASCAT   8821.0    INPP4B     DBSNP      O15327  \n",
       "2              1         GWASCAT   8821.0    INPP4B       VEP      O15327  \n",
       "3              1         CLINVAR   3508.0   IGHMBP2       VEP      P38935  \n",
       "4              1         CLINVAR   3508.0   IGHMBP2     DBSNP      P38935  \n",
       "...          ...             ...      ...       ...       ...         ...  \n",
       "489665         1         CLINVAR  54714.0     CNGB3       VEP      Q9NQW8  \n",
       "489666         1  GWASCAT;GWASDB   3710.0     ITPR3       VEP      Q14573  \n",
       "489667         1  GWASCAT;GWASDB   3710.0     ITPR3     DBSNP      Q14573  \n",
       "489668         1         GWASCAT      NaN       NaN       NaN         NaN  \n",
       "489669         1         GWASCAT   4887.0     NPY2R     DBSNP      P49146  \n",
       "\n",
       "[489670 rows x 20 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "disgenet_total = vda.merge(variant_gene, on=['snp_id'], how='left')\n",
    "disgenet_total"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['snp_id', 'chromosome', 'position', 'dsi', 'dpi', 'diseaseid',\n",
       "       'diseasename', 'diseasetype', 'diseaseclass', 'diseasesemantictype',\n",
       "       'score', 'ei', 'yearinitial', 'yearfinal', 'nofpmids', 'source',\n",
       "       'gene_id', 'gene_name', 'source_id', 'uniprot_acc'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "disgenet_total.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Dividir la tabla total en distintos subsets: Genes; Diseases; Stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "disgenet_total_genes = disgenet_total[['snp_id', 'uniprot_acc', 'gene_name', 'gene_id', 'chromosome', 'position', 'source']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "disgenet_total_disease = disgenet_total[['snp_id', 'gene_name', 'gene_id', 'diseaseid', 'diseasename', 'diseasetype', 'diseaseclass', 'diseasesemantictype',\n",
    "                                        'source']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snp_id</th>\n",
       "      <th>uniprot_acc</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs1000005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>33060745</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>O15327</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>8821.0</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs10000770</td>\n",
       "      <td>O15327</td>\n",
       "      <td>INPP4B</td>\n",
       "      <td>8821.0</td>\n",
       "      <td>4</td>\n",
       "      <td>142693109</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>P38935</td>\n",
       "      <td>IGHMBP2</td>\n",
       "      <td>3508.0</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>CLINVAR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1000091588</td>\n",
       "      <td>P38935</td>\n",
       "      <td>IGHMBP2</td>\n",
       "      <td>3508.0</td>\n",
       "      <td>11</td>\n",
       "      <td>68935374</td>\n",
       "      <td>CLINVAR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489665</th>\n",
       "      <td>rs999921351</td>\n",
       "      <td>Q9NQW8</td>\n",
       "      <td>CNGB3</td>\n",
       "      <td>54714.0</td>\n",
       "      <td>8</td>\n",
       "      <td>86628939</td>\n",
       "      <td>CLINVAR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489666</th>\n",
       "      <td>rs999943</td>\n",
       "      <td>Q14573</td>\n",
       "      <td>ITPR3</td>\n",
       "      <td>3710.0</td>\n",
       "      <td>6</td>\n",
       "      <td>33656956</td>\n",
       "      <td>GWASCAT;GWASDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489667</th>\n",
       "      <td>rs999943</td>\n",
       "      <td>Q14573</td>\n",
       "      <td>ITPR3</td>\n",
       "      <td>3710.0</td>\n",
       "      <td>6</td>\n",
       "      <td>33656956</td>\n",
       "      <td>GWASCAT;GWASDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489668</th>\n",
       "      <td>rs999944</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>64822719</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489669</th>\n",
       "      <td>rs9999820</td>\n",
       "      <td>P49146</td>\n",
       "      <td>NPY2R</td>\n",
       "      <td>4887.0</td>\n",
       "      <td>4</td>\n",
       "      <td>155197173</td>\n",
       "      <td>GWASCAT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>489670 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              snp_id uniprot_acc gene_name  gene_id chromosome   position  \\\n",
       "0          rs1000005         NaN       NaN      NaN         21   33060745   \n",
       "1         rs10000770      O15327    INPP4B   8821.0          4  142693109   \n",
       "2         rs10000770      O15327    INPP4B   8821.0          4  142693109   \n",
       "3       rs1000091588      P38935   IGHMBP2   3508.0         11   68935374   \n",
       "4       rs1000091588      P38935   IGHMBP2   3508.0         11   68935374   \n",
       "...              ...         ...       ...      ...        ...        ...   \n",
       "489665   rs999921351      Q9NQW8     CNGB3  54714.0          8   86628939   \n",
       "489666      rs999943      Q14573     ITPR3   3710.0          6   33656956   \n",
       "489667      rs999943      Q14573     ITPR3   3710.0          6   33656956   \n",
       "489668      rs999944         NaN       NaN      NaN          2   64822719   \n",
       "489669     rs9999820      P49146     NPY2R   4887.0          4  155197173   \n",
       "\n",
       "                source  \n",
       "0              GWASCAT  \n",
       "1              GWASCAT  \n",
       "2              GWASCAT  \n",
       "3              CLINVAR  \n",
       "4              CLINVAR  \n",
       "...                ...  \n",
       "489665         CLINVAR  \n",
       "489666  GWASCAT;GWASDB  \n",
       "489667  GWASCAT;GWASDB  \n",
       "489668         GWASCAT  \n",
       "489669         GWASCAT  \n",
       "\n",
       "[489670 rows x 7 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "disgenet_total_genes"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
