{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5774d842",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "\n",
    "p_prot = pd.read_csv(\"../db_tables/protein.tsv\", sep=\"\\t\")\n",
    "p_prot = p_prot[['id_protein', 'uniprot_acc']]\n",
    "\n",
    "#by ensemble transcript\n",
    "protein_enst = pd.read_csv(\"../raw_data/llps_uniprot2ENST.tab.txt\", sep=\"\\t\")\n",
    "protein_enst = protein_enst.rename(columns={'From': 'uniprot_acc', 'To':'ensembl_enst'})\n",
    "protein_enst = protein_enst.merge(p_prot[['id_protein', 'uniprot_acc']]).drop(columns=['uniprot_acc'])\n",
    "\n",
    "#by ensemble protein\n",
    "protein_ensp = pd.read_csv(\"../raw_data/llps_uniprot2ENSP.tab.txt\", sep=\"\\t\")\n",
    "protein_ensp = protein_ensp.rename(columns={'From': 'uniprot_acc', 'To':'ensembl_ensp'})\n",
    "protein_ensp = protein_ensp.merge(p_prot[['id_protein', 'uniprot_acc']]).drop(columns=['uniprot_acc'])\n",
    "\n",
    "datafile  = \"../raw_data/cosmic_v94/CosmicMutantExport_sel.tsv\"\n",
    "#for each sample_id has the cancer types columns and the cosmic_phenotype_id\n",
    "samples_data = \"../raw_data/cosmic_v94/CosmicSample.tsv\" \n",
    "#for each cosmic_phenotype_id has the NCI_CODE and other like EFO, MONDO, ....\n",
    "class_data = \"../raw_data/cosmic_v94/classification.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6339c492",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['id_sample', 'GENOMIC_MUTATION_ID', 'LEGACY_MUTATION_ID', 'MUTATION_ID', 'CDS', 'AA', 'consequence', 'genomic', 'somatic_status', 'pubmed', 'ensembl_ensp', 'ensembl_enst']\n",
      "   id_sample GENOMIC_MUTATION_ID LEGACY_MUTATION_ID  MUTATION_ID  \\\n",
      "0    2198328        COSV57322711        COSM1628357     98293492   \n",
      "1    2869867        COSV57321510        COSM1706236     98292376   \n",
      "2    2549254        COSV57321745        COSM5796874     98292486   \n",
      "3    2749203        COSV57324149        COSM7003726     98292523   \n",
      "4    2839416       COSV104562760        COSM9463514     98293844   \n",
      "\n",
      "                 CDS       AA              consequence  \\\n",
      "0           c.609G>T  p.M203I  Substitution - Missense   \n",
      "1           c.781G>A  p.E261K  Substitution - Missense   \n",
      "2           c.385G>C  p.G129R  Substitution - Missense   \n",
      "3            c.20G>C    p.G7A  Substitution - Missense   \n",
      "4  c.559_560delinsAA  p.G187K  Substitution - Missense   \n",
      "\n",
      "                  genomic             somatic_status    pubmed  \\\n",
      "0  12:108894364-108894364  Confirmed somatic variant       NaN   \n",
      "1  12:108898764-108898764  Confirmed somatic variant  26950094   \n",
      "2  12:108889544-108889544  Confirmed somatic variant       NaN   \n",
      "3  12:108885026-108885026  Confirmed somatic variant  29253671   \n",
      "4  12:108894314-108894315  Confirmed somatic variant  28467829   \n",
      "\n",
      "      ensembl_ensp     ensembl_enst  \n",
      "0  ENSP00000228476  ENST00000228476  \n",
      "1  ENSP00000228476  ENST00000228476  \n",
      "2  ENSP00000228476  ENST00000228476  \n",
      "3  ENSP00000228476  ENST00000228476  \n",
      "4  ENSP00000228476  ENST00000228476  \n",
      "cosmic rows 6795791\n"
     ]
    }
   ],
   "source": [
    "mutation = pd.read_csv(datafile, sep=\"\\t\")\n",
    "#to string the columns\n",
    "mutation['pubmed'] = mutation['pubmed'].map(lambda x: str(int(x)) if not np.isnan(x) else x)\n",
    "print(list(mutation.columns))\n",
    "print(mutation.head())\n",
    "print(f'cosmic rows {mutation.shape[0]}') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7bfdc7d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         ensembl_enst  counts\n",
      "3692  ENST00000529639       2\n",
      "Empty DataFrame\n",
      "Columns: [ensembl_ensp, counts]\n",
      "Index: []\n",
      "rows mapped by ENSP_name 0\n",
      "rows mapped by ENST_name 1724388\n",
      "cosmic without duplicates 1724388\n",
      "cosmic proteins 4018\n"
     ]
    }
   ],
   "source": [
    "# Filter only mutations in our LLPS dataset \n",
    "\n",
    "mutation['internalid'] = range(1, len(mutation)+1)\n",
    "\n",
    "#merge by order 'ensembl_enst', 'ensembl_ensp'\n",
    "#to avoid the rows duplication\n",
    "protein_enst = mutation.merge(protein_enst)\n",
    "\n",
    "y =  protein_enst[['ensembl_enst', 'id_protein']]\n",
    "y = y.drop_duplicates()\n",
    "dd = y.groupby(['ensembl_enst']).size().reset_index(name='counts')\n",
    "print(dd[dd.counts > 1])\n",
    "#este es un caso raro ya que la secuencia del transcripto es la union de dos uniprots\n",
    "#P35544 + P62861\n",
    "#id_protein 4340 y 4342\n",
    "\n",
    "mapped_internal = protein_enst['internalid'].unique().tolist()\n",
    "mutation = mutation[~mutation['internalid'].isin(mapped_internal)]\n",
    "protein_ensp = mutation.merge(protein_ensp)\n",
    "y =  protein_ensp[['ensembl_ensp', 'id_protein']]\n",
    "y = y.drop_duplicates()\n",
    "dd = y.groupby(['ensembl_ensp']).size().reset_index(name='counts')\n",
    "print(dd[dd.counts > 1])\n",
    "\n",
    "print(f'rows mapped by ENSP_name {protein_ensp.shape[0]}')\n",
    "print(f'rows mapped by ENST_name {protein_enst.shape[0]}')\n",
    "\n",
    "mutation = pd.concat([protein_enst, protein_ensp], ignore_index=True)\n",
    "mutation = mutation.drop_duplicates()\n",
    "print(f'cosmic without duplicates {mutation.shape[0]}') \n",
    "print(f'cosmic proteins {len(mutation[\"id_protein\"].unique())}')                \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ca383815",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1144756\n",
      "        LEGACY_MUTATION_ID  counts\n",
      "8381           COSM1049194       2\n",
      "8382           COSM1049196       2\n",
      "33704          COSM1206683       2\n",
      "59817          COSM1355890       2\n",
      "85748           COSM160375       2\n",
      "...                    ...     ...\n",
      "1063915        COSM9462306       2\n",
      "1071035        COSM9508187       2\n",
      "1084998        COSM9589314       2\n",
      "1090340        COSM9621560       2\n",
      "1099092         COSM967422       2\n",
      "\n",
      "[81 rows x 2 columns]\n",
      "Empty DataFrame\n",
      "Columns: [LEGACY_MUTATION_ID, id_protein, counts]\n",
      "Index: []\n"
     ]
    }
   ],
   "source": [
    "#ver si hay mutiples transcriptos de una misma proteina para eliminarlos \n",
    "#un legacy_mutation_id tiene todas los transcriptos mutados que corresponde a una misma posicion\n",
    "y = mutation[['LEGACY_MUTATION_ID', 'id_protein']]\n",
    "y = y.drop_duplicates()\n",
    "print(y.shape[0])\n",
    "mut_trans = y.groupby(['LEGACY_MUTATION_ID']).size().reset_index(name='counts')\n",
    "mut_trans = mut_trans[mut_trans.counts > 1]\n",
    "print(mut_trans)\n",
    "#puede darse el caso que tengan varios protein_id ya que los uniprot asociados son disitntas variantes de splicing\n",
    "y = mutation[['LEGACY_MUTATION_ID', 'id_protein', 'ensembl_enst']]\n",
    "y = y.drop_duplicates()\n",
    "mut_trans = y.groupby(['LEGACY_MUTATION_ID', 'id_protein']).size().reset_index(name='counts')\n",
    "mut_trans = mut_trans[mut_trans.counts > 1]\n",
    "print(mut_trans)\n",
    "#only one transcript per protein id\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6456a9ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample-mutation 1723613\n",
      "mutation-idprot-transcr 1144756\n"
     ]
    }
   ],
   "source": [
    "#ver si id_mutation and id_protein es unique de la tabla\n",
    "#de ser asi, dejar solo id_mutation y datos de las mutaciones juntas\n",
    "#y otra tabla con \"ID_sample\", \"GENOMIC_MUTATION_ID\", \"LEGACY_MUTATION_ID\", \"MUTATION_ID\", \"Mutation somatic status\", \"Pubmed_PMID\"\n",
    "\n",
    "samples_mut = mutation[[\"id_sample\", \"LEGACY_MUTATION_ID\", 'id_protein', \"somatic_status\", \"pubmed\"]]\n",
    "samples_mut = samples_mut.drop_duplicates()\n",
    "print(f'sample-mutation {samples_mut.shape[0]}')\n",
    "#samples_mut.to_csv('samples_mut.tsv', sep=\"\\t\", index= False)\n",
    "#\"LEGACY_MUTATION_ID\" + 'id_protein' is the mutation id\n",
    "mutation = mutation[[\"LEGACY_MUTATION_ID\", 'id_protein', 'CDS', 'AA', 'consequence', 'genomic']]\n",
    "mutation = mutation.drop_duplicates()\n",
    "print(f'mutation-idprot-transcr {mutation.shape[0]}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0ca7b3e1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['10', '2', '15', '11', '19', '7', '17', '1', '3', '12', '14', '5', '9', '4', '16', '6', '22', '21', '18', '8', '20', '23', '13', '24', '25']\n"
     ]
    }
   ],
   "source": [
    "#split genomic by (x:y-z) in chromosome, start_genomic end_genomic\n",
    "mutation['genomic'] = mutation['genomic'].str.findall('^(.+):(.+)-(.+)$').str[0]\n",
    "mutation['chromosome'] = mutation['genomic'].map(lambda x: x[0])\n",
    "mutation['start_genomic'] = mutation['genomic'].map(lambda x: x[1])\n",
    "mutation['end_genomic'] = mutation['genomic'].map(lambda x: x[2])\n",
    "mutation = mutation.drop(columns=['genomic'])\n",
    "print(mutation['chromosome'].unique().tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "424c7338",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1144756\n",
      "1144756\n",
      "        LEGACY_MUTATION_ID  id_protein  \\\n",
      "0               COSM166059        1011   \n",
      "2              COSM6982134        1011   \n",
      "3               COSM428012        1011   \n",
      "5               COSM427996        1011   \n",
      "6              COSM5703736        1011   \n",
      "...                    ...         ...   \n",
      "1724382        COSM7772448        1844   \n",
      "1724383        COSM4622984        1844   \n",
      "1724384        COSM9330544        1844   \n",
      "1724386        COSM8832530        1844   \n",
      "1724387        COSM3944131        1844   \n",
      "\n",
      "                                                       CDS  \\\n",
      "0                                                  1224dup   \n",
      "2        661_662delinsCCTGCCTGTTGTGAGCTGCTCTACGTGCCCTAC...   \n",
      "3                                             1302_1303dup   \n",
      "5                                                  1203dup   \n",
      "6                                                   633del   \n",
      "...                                                    ...   \n",
      "1724382                                             134C>T   \n",
      "1724383                                             366T>C   \n",
      "1724384                                             124T>C   \n",
      "1724386                                              82G>T   \n",
      "1724387                                               5C>G   \n",
      "\n",
      "                              AA            consequence_cosmic chromosome  \\\n",
      "0                     P409Afs*99        Insertion - Frameshift         10   \n",
      "2        T221delinsPACCELLYVPYVL                           NaN         10   \n",
      "3                     H435Pfs*42        Insertion - Frameshift         10   \n",
      "5                    S402Vfs*106        Insertion - Frameshift         10   \n",
      "6                     S212Pfs*54         Deletion - Frameshift         10   \n",
      "...                          ...                           ...        ...   \n",
      "1724382                     P45L       Substitution - Missense          1   \n",
      "1724383                    D122=  Substitution - coding silent          1   \n",
      "1724384                     F42L       Substitution - Missense          1   \n",
      "1724386                     D28Y       Substitution - Missense          1   \n",
      "1724387                      S2C       Substitution - Missense          1   \n",
      "\n",
      "        start_genomic end_genomic  indexinternal  \n",
      "0             8073912     8073913              1  \n",
      "2             8058724     8058725              2  \n",
      "3             8073991     8073992              3  \n",
      "5             8073891     8073892              4  \n",
      "6             8058696     8058696              5  \n",
      "...               ...         ...            ...  \n",
      "1724382      45515780    45515780        1144752  \n",
      "1724383      45514890    45514890        1144753  \n",
      "1724384      45515790    45515790        1144754  \n",
      "1724386      45518962    45518962        1144755  \n",
      "1724387      45519039    45519039        1144756  \n",
      "\n",
      "[1144756 rows x 9 columns]\n"
     ]
    }
   ],
   "source": [
    "print(mutation.shape[0])\n",
    "mutation = mutation.rename(columns={'consequence': 'consequence_cosmic'})\n",
    "mutation['indexinternal'] = range(1, len(mutation)+1)\n",
    "mutation['AA'] = mutation.AA.map(lambda x: re.findall('^p\\.(.*)$', x))\n",
    "mutation['AA'] = mutation.AA.str[0]\n",
    "mutation = mutation[mutation.AA.notnull()]\n",
    "\n",
    "mutation['CDS'] = mutation.CDS.map(lambda x: re.findall('^c\\.(.*)$', x))\n",
    "mutation['CDS'] = mutation.CDS.str[0]\n",
    "mutation = mutation[mutation.CDS.notnull()]\n",
    "\n",
    "print(mutation.shape[0])\n",
    "print(mutation)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cad313a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def separar_en_cols(df, column, conseq, conseq_regex, override=False):\n",
    "    '''\n",
    "    recibe un DataFrame, el nombre de una columna auxiliar (column)\n",
    "    y un string con el tipo de consecuencia (conseq). La col. auxiliar\n",
    "    es una tupla con los elementos implicados en una mutacion\n",
    "    como la siguiente (aa1, start_pos, aa2, end_pos, aa/s_nuevos).\n",
    "    Devuelve el DataFrame df con estas 5 nuevas columnas\n",
    "    '''\n",
    "    \n",
    "    df_crop = df.copy()\n",
    "      \n",
    "      \n",
    "    if override:\n",
    "        df_crop['aux'] = df_crop[column].str.findall(conseq_regex).str[0]\n",
    "    else:\n",
    "        df_crop['aux'] = df_crop[column].str.findall('^([A-Z\\*]+)(\\d+)_?([A-Z\\*])?(\\d+)?'+conseq_regex+'(.*)$').str[0]\n",
    "  \n",
    "    df_crop = df_crop[~df_crop['aux'].isnull()]\n",
    "    '''\n",
    "    #check list type bad format    \n",
    "    if conseq in ['synonym', 'nonsense']: #(w, d, '', '', '')\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if ((x[0] == '') or (x[1] == '') or (x[2] != '') or (x[3] != '') or (x[4] != '')) else False)\n",
    "    elif conseq in ['delins']: #(w, d, '', '', w) or (w, d, w, d, w)\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if (((x[0] == '') or (x[1] == '') or (x[4] == '')) or ((x[2] != '') and (x[3] == '')) or ((x[2] == '') and (x[3] != ''))) else False) \n",
    "    elif conseq in ['deletion', 'duplication']: #(w, d, '', '', '') or (w, d, w, d, '')\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if (((x[0] == '') or (x[1] == '') or (x[4] != '')) or ((x[2] != '') and (x[3] == '')) or ((x[2] == '') and (x[3] != ''))) else False) \n",
    "    elif conseq in ['insertion']: #(w, d, w, d, w)\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if ((x[0] == '') or (x[1] == '') or (x[2] == '') or (x[3] == '') or (x[4] == '')) else False) \n",
    "    elif conseq in [\"frameshift\", 'nonstop']: # (w, d, w, '', w) or (w, d, w, '', '')\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if ((x[0] == '') or (x[1] == '') or (x[2] == '') or (x[3] != '')) else False) \n",
    "        #bad format in nonstop for the delext* (w, d, '', '', w) or (w, d, '', '','')\n",
    "    elif conseq in ['missense']: # (w, d, w, '', '') \n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if ((x[0] == '') or (x[1] == '') or (x[2] == '') or (x[3] != '') or (x[4] != '')) else False)  \n",
    "    '''\n",
    "    # start position\n",
    "    df_crop['start_aa'] = df_crop['aux'].map(lambda x: x[1])\n",
    "    # end position\n",
    "    df_crop['end_aa'] = df_crop['aux'].map(lambda x: x[3] if x[3] != '' else x[1]) # poner en el end el start si no hay end\n",
    "        \n",
    "    # from: es el/los aa que cambian # concateno si existe mas de un aa que cambia (o sea, si es un rango)\n",
    "    df_crop['from_aa'] = df_crop['aux'].map(lambda x: str(x[0]) + ('-' + str(x[2]) if x[2] != '' and x[3] != '' else '')) \n",
    "        \n",
    "    if conseq == \"missense\":     \n",
    "        df_crop['to_aa'] = df_crop['aux'].map(lambda x: x[2])\n",
    "    elif conseq == \"nonsense\":\n",
    "        df_crop['to_aa'] = \"*\"\n",
    "    elif conseq in [\"frameshift\", 'nonstop']:    \n",
    "        df_crop['to_aa'] = \"\"\n",
    "    else: # to: aa/s nuevos\n",
    "        df_crop['to_aa'] = df_crop['aux'].map(lambda x: x[4])\n",
    "\n",
    "    # consecuencia de la mutacion\n",
    "    df_crop['consequence_protein'] = conseq\n",
    "    #print(df_crop[df_crop['badformat'] == True].sort_values(by='AA', key=lambda col: col.str.len(), ascending=False, na_position='last'))\n",
    "    \n",
    "    df_crop = df_crop.drop(columns=['aux'])\n",
    "\n",
    "    return df_crop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0246f498",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all mutations 1144756\n",
      "Found 264334 synonym\n",
      "Found 1606 delins\n",
      "Found 5403 deletions\n",
      "Found 884 inserions\n",
      "Found 38326 frameshift\n",
      "Found 1031 duplications\n",
      "Found 749 nonstop\n",
      "Found 60124 nonsense\n",
      "Found 771186 missense\n",
      "mapped 1143643 mutations \n",
      "No mapped 1113 mutations\n"
     ]
    }
   ],
   "source": [
    "# A DataFrame for each molecular consequence\n",
    "print(\"Mapping AA mutations\")\n",
    "mutation_used = mutation.copy()\n",
    "print(f\"all mutations {mutation.shape[0]}\")\n",
    "\n",
    "synonym = separar_en_cols(mutation_used, \"AA\", \"synonym\", \"=$\")\n",
    "print(f\"Found {synonym.shape[0]} synonym\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(synonym['indexinternal'].tolist())]\n",
    "delins = separar_en_cols(mutation_used, \"AA\", \"delins\", \"delins\")\n",
    "print(f\"Found {delins.shape[0]} delins\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(delins['indexinternal'].tolist())]\n",
    "deletions = separar_en_cols(mutation_used, \"AA\", \"deletion\", \"del$\") # finish with 'del'\n",
    "print(f\"Found {deletions.shape[0]} deletions\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(deletions['indexinternal'].tolist())]\n",
    "insertions = separar_en_cols(mutation_used, \"AA\", \"insertion\", \"ins\")\n",
    "print(f\"Found {insertions.shape[0]} inserions\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(insertions['indexinternal'].tolist())]\n",
    "frameshift = separar_en_cols(mutation_used, \"AA\", \"frameshift\", 'fs\\*')\n",
    "print(f\"Found {frameshift.shape[0]} frameshift\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(frameshift['indexinternal'].tolist())]\n",
    "duplications = separar_en_cols(mutation_used, \"AA\", \"duplication\", \"dup\")\n",
    "print(f\"Found {duplications.shape[0]} duplications\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(duplications['indexinternal'].tolist())]\n",
    "nonstop = separar_en_cols(mutation_used, \"AA\", \"nonstop\", '(?:del)?ext\\*')\n",
    "print(f\"Found {nonstop.shape[0]} nonstop\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(nonstop['indexinternal'].tolist())]\n",
    "nonsense = separar_en_cols(mutation_used, \"AA\", \"nonsense\", \"\\*$\")\n",
    "print(f\"Found {nonsense.shape[0]} nonsense\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(nonsense['indexinternal'].tolist())]\n",
    "missense = separar_en_cols(mutation_used, \"AA\", \"missense\", '^([A-Z])(\\d+)([A-Z])$(.*)?(.*)?(.*)$', override=True)\n",
    "print(f\"Found {missense.shape[0]} missense\")\n",
    "\n",
    "#los frameshift M1?, M1_?\\d*\n",
    "## Concatenate everything\n",
    "mutation_mapped = pd.concat([synonym, delins, deletions, insertions, frameshift, duplications, nonstop, nonsense, missense])\n",
    "\n",
    "print(f\"mapped {mutation_mapped.shape[0]} mutations \")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(mutation_mapped['indexinternal'].tolist())]\n",
    "print(f\"No mapped {mutation_used.shape[0]} mutations\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "e486245f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def separar_en_cols_cds(df, column, conseq, conseq_regex, override=False):\n",
    "    '''\n",
    "    recibe un DataFrame, el nombre de una columna auxiliar (column)\n",
    "    y un string con el tipo de consecuencia (conseq). La col. auxiliar\n",
    "    es una tupla con los elementos implicados en una mutacion\n",
    "    como la siguiente (nucleatides1, nucleatides2).\n",
    "    Devuelve el DataFrame df con estas 2 nuevas columnas\n",
    "    '''\n",
    "    \n",
    "    df_crop = df.copy()\n",
    "      \n",
    "      \n",
    "    if override:\n",
    "        df_crop['aux'] = df_crop[column].str.findall(conseq_regex).str[0]\n",
    "    else:\n",
    "        df_crop['aux'] = df_crop[column].str.findall('^[-\\+\\d]+_?(?:[-\\+\\d]+)?'+conseq_regex+'([ACTG]+)?>?([ACTG]+)?$').str[0]\n",
    "  \n",
    "    df_crop = df_crop[~df_crop['aux'].isnull()]\n",
    "    \n",
    "    df_crop['from_genomic'] = ''\n",
    "    df_crop['to_genomic'] = ''\n",
    "    \n",
    "    if conseq in [\"delins\", 'insertion']: \n",
    "        df_crop['to_genomic'] = df_crop['aux'].map(lambda x: x[0])\n",
    "    elif conseq == 'missense': \n",
    "        df_crop['from_genomic'] = df_crop['aux'].map(lambda x: x[0])\n",
    "        df_crop['to_genomic'] = df_crop['aux'].map(lambda x: x[1])\n",
    "\n",
    "    # consecuencia de la mutacion\n",
    "    df_crop['consequence_genomic'] = conseq\n",
    "    #print(df_crop[df_crop['badformat'] == True].sort_values(by='AA', key=lambda col: col.str.len(), ascending=False, na_position='last'))\n",
    "    \n",
    "    df_crop = df_crop.drop(columns=['aux'])\n",
    "\n",
    "    return df_crop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "3fc48fba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all mutations 1143643\n",
      "Found 9179 delins cds\n",
      "Found 31129 deletions cds\n",
      "Found 4077 inserions cds\n",
      "Found 12871 duplications cds\n",
      "Found 320 invertion cds\n",
      "Found 1086067 missense cds\n",
      "mapped 1143643 mutations \n",
      "No mapped 0 mutations\n"
     ]
    }
   ],
   "source": [
    "print(\"Mapping CDS mutations\")\n",
    "\n",
    "mutation_used = mutation_mapped.copy()\n",
    "print(f\"all mutations {mutation_used.shape[0]}\")\n",
    "\n",
    "delins = separar_en_cols_cds(mutation_used, \"CDS\", \"delins\", \"delins\")\n",
    "print(f\"Found {delins.shape[0]} delins cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(delins['indexinternal'].tolist())]\n",
    "deletions = separar_en_cols_cds(mutation_used, \"CDS\", \"deletion\", \"del$\") # finish with 'del'\n",
    "print(f\"Found {deletions.shape[0]} deletions cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(deletions['indexinternal'].tolist())]\n",
    "insertions = separar_en_cols_cds(mutation_used, \"CDS\", \"insertion\", \"ins\")\n",
    "print(f\"Found {insertions.shape[0]} inserions cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(insertions['indexinternal'].tolist())]\n",
    "duplications = separar_en_cols_cds(mutation_used, \"CDS\", \"duplication\", \"dup$\")\n",
    "print(f\"Found {duplications.shape[0]} duplications cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(duplications['indexinternal'].tolist())]\n",
    "inversions = separar_en_cols_cds(mutation_used, \"CDS\", \"invertion\", \"inv$\")\n",
    "print(f\"Found {invertions.shape[0]} invertion cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(inversions['indexinternal'].tolist())]\n",
    "substitution = separar_en_cols_cds(mutation_used, \"CDS\", \"missense\", '')\n",
    "print(f\"Found {missense.shape[0]} missense cds\")\n",
    "\n",
    "mutation_mapped2 = pd.concat([delins, deletions, insertions, duplications, inversions, substitution])\n",
    "\n",
    "print(f\"mapped {mutation_mapped2.shape[0]} mutations \")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(mutation_mapped2['indexinternal'].tolist())]\n",
    "print(f\"No mapped {mutation_used.shape[0]} mutations\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "d2cc4df2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        LEGACY_MUTATION_ID  id_protein                CDS      AA  \\\n",
      "6221           COSM9650587         285  2304_2305delinsTT  FL768=   \n",
      "11001           COSM398992        3471  1044_1045delinsTT  LL348=   \n",
      "36343          COSM1293677        1923    111_112delinsTT   LL37=   \n",
      "36479          COSM9649600        1496    351_352delinsTT  IL117=   \n",
      "42720          COSM9455811        1912  1749_1750delinsTT  LL583=   \n",
      "...                    ...         ...                ...     ...   \n",
      "1724381        COSM9416203        1844             163G>A    E55K   \n",
      "1724382        COSM7772448        1844             134C>T    P45L   \n",
      "1724384        COSM9330544        1844             124T>C    F42L   \n",
      "1724386        COSM8832530        1844              82G>T    D28Y   \n",
      "1724387        COSM3944131        1844               5C>G     S2C   \n",
      "\n",
      "                   consequence_cosmic chromosome start_genomic end_genomic  \\\n",
      "6221     Substitution - coding silent          3     124398835   124398836   \n",
      "11001    Substitution - coding silent         19       4446289     4446290   \n",
      "36343    Substitution - coding silent         17      42562646    42562647   \n",
      "36479    Substitution - coding silent          6      31616498    31616499   \n",
      "42720    Substitution - coding silent         10      60196565    60196566   \n",
      "...                               ...        ...           ...         ...   \n",
      "1724381       Substitution - Missense          1      45515751    45515751   \n",
      "1724382       Substitution - Missense          1      45515780    45515780   \n",
      "1724384       Substitution - Missense          1      45515790    45515790   \n",
      "1724386       Substitution - Missense          1      45518962    45518962   \n",
      "1724387       Substitution - Missense          1      45519039    45519039   \n",
      "\n",
      "         indexinternal start_aa end_aa from_aa to_aa consequence_protein  \\\n",
      "6221              4304      768    768      FL                   synonym   \n",
      "11001             8061      348    348      LL                   synonym   \n",
      "36343            26499       37     37      LL                   synonym   \n",
      "36479            26613      117    117      IL                   synonym   \n",
      "42720            31452      583    583      LL                   synonym   \n",
      "...                ...      ...    ...     ...   ...                 ...   \n",
      "1724381        1144751       55     55       E     K            missense   \n",
      "1724382        1144752       45     45       P     L            missense   \n",
      "1724384        1144754       42     42       F     L            missense   \n",
      "1724386        1144755       28     28       D     Y            missense   \n",
      "1724387        1144756        2      2       S     C            missense   \n",
      "\n",
      "        from_genomic to_genomic consequence_genomic  \n",
      "6221                         TT              delins  \n",
      "11001                        TT              delins  \n",
      "36343                        TT              delins  \n",
      "36479                        TT              delins  \n",
      "42720                        TT              delins  \n",
      "...              ...        ...                 ...  \n",
      "1724381            G          A            missense  \n",
      "1724382            C          T            missense  \n",
      "1724384            T          C            missense  \n",
      "1724386            G          T            missense  \n",
      "1724387            C          G            missense  \n",
      "\n",
      "[1143643 rows x 17 columns]\n"
     ]
    }
   ],
   "source": [
    "print(mutation_mapped2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2fd6f182",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
