{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8977faff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Ala542_Lysext-5', 'Ala542_Lys543', 'Ala542_Lysfs', 'Ala542_Lys543insAlaLysAlaAla']\n",
      "['A542_Kext-5', 'A542_K543', 'A542_Kfs', 'A542_K543insAKAA']\n",
      "['A542_Kext-5', 'A542_K543', 'A542_Kfs', 'A542_K543insAKAA']\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "y = [\"Ala542_Lysext-5\", \"Ala542_Lys543\", \"Ala542_Lysfs\", \"Ala542_Lys543insAlaLysAlaAla\"]\n",
    "print(y)\n",
    "d = {\"Ala\": \"A\", \"Lys\": \"K\"}\n",
    "def replace_by_dict (mut, dict_aa):\n",
    "    for k,v in dict_aa.items():\n",
    "        mut = mut.replace(k,v)\n",
    "    return mut\n",
    "y2 = [replace_by_dict(mut, d) for mut in y]\n",
    "print(y2)\n",
    "\n",
    "\n",
    "# Create a regular expression from all of the dictionary keys\n",
    "regex = re.compile(\"|\".join(map(re.escape, d.keys())))\n",
    "# For each match, look up the corresponding value in the dictionary\n",
    "y3 = [regex.sub(lambda match: d[match.group(0)], mut) for mut in y]\n",
    "print(y3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5774d842",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "\n",
    "p_prot = pd.read_csv(\"../db_tables/protein.tsv\", sep=\"\\t\")\n",
    "p_prot = p_prot[['id_protein', 'uniprot_acc']]\n",
    "\n",
    "#by ensemble transcript\n",
    "protein_enst = pd.read_csv(\"../raw_data/llps_uniprot2ENST.tab.txt\", sep=\"\\t\")\n",
    "protein_enst = protein_enst.rename(columns={'From': 'uniprot_acc', 'To':'ensembl_enst'})\n",
    "protein_enst = protein_enst.merge(p_prot[['id_protein', 'uniprot_acc']]).drop(columns=['uniprot_acc'])\n",
    "\n",
    "#by ensemble protein\n",
    "protein_ensp = pd.read_csv(\"../raw_data/llps_uniprot2ENSP.tab.txt\", sep=\"\\t\")\n",
    "protein_ensp = protein_ensp.rename(columns={'From': 'uniprot_acc', 'To':'ensembl_ensp'})\n",
    "protein_ensp = protein_ensp.merge(p_prot[['id_protein', 'uniprot_acc']]).drop(columns=['uniprot_acc'])\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "datafile  = \"../raw_data/cosmic_v94/CosmicMutantExport_sel.tsv\"\n",
    "#for each sample_id has the cancer types columns and the cosmic_phenotype_id\n",
    "samples_data = \"../raw_data/cosmic_v94/CosmicSample.tsv\" \n",
    "#for each cosmic_phenotype_id has the NCI_CODE and other like EFO, MONDO, ....\n",
    "class_data = \"../raw_data/cosmic_v94/classification.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6339c492",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['id_sample', 'GENOMIC_MUTATION_ID', 'LEGACY_MUTATION_ID', 'MUTATION_ID', 'CDS', 'AA', 'consequence', 'genomic', 'somatic_status', 'pubmed', 'ensembl_ensp', 'ensembl_enst']\n",
      "   id_sample GENOMIC_MUTATION_ID LEGACY_MUTATION_ID  MUTATION_ID  \\\n",
      "0    2198328        COSV57322711        COSM1628357     98293492   \n",
      "1    2869867        COSV57321510        COSM1706236     98292376   \n",
      "2    2549254        COSV57321745        COSM5796874     98292486   \n",
      "3    2749203        COSV57324149        COSM7003726     98292523   \n",
      "4    2839416       COSV104562760        COSM9463514     98293844   \n",
      "\n",
      "                 CDS       AA              consequence  \\\n",
      "0           c.609G>T  p.M203I  Substitution - Missense   \n",
      "1           c.781G>A  p.E261K  Substitution - Missense   \n",
      "2           c.385G>C  p.G129R  Substitution - Missense   \n",
      "3            c.20G>C    p.G7A  Substitution - Missense   \n",
      "4  c.559_560delinsAA  p.G187K  Substitution - Missense   \n",
      "\n",
      "                  genomic             somatic_status    pubmed  \\\n",
      "0  12:108894364-108894364  Confirmed somatic variant       NaN   \n",
      "1  12:108898764-108898764  Confirmed somatic variant  26950094   \n",
      "2  12:108889544-108889544  Confirmed somatic variant       NaN   \n",
      "3  12:108885026-108885026  Confirmed somatic variant  29253671   \n",
      "4  12:108894314-108894315  Confirmed somatic variant  28467829   \n",
      "\n",
      "      ensembl_ensp     ensembl_enst  \n",
      "0  ENSP00000228476  ENST00000228476  \n",
      "1  ENSP00000228476  ENST00000228476  \n",
      "2  ENSP00000228476  ENST00000228476  \n",
      "3  ENSP00000228476  ENST00000228476  \n",
      "4  ENSP00000228476  ENST00000228476  \n",
      "cosmic rows 6795791\n"
     ]
    }
   ],
   "source": [
    "mutation = pd.read_csv(datafile, sep=\"\\t\")\n",
    "#to string the columns\n",
    "mutation['pubmed'] = mutation['pubmed'].map(lambda x: str(int(x)) if not np.isnan(x) else x)\n",
    "print(list(mutation.columns))\n",
    "print(mutation.head())\n",
    "print(f'cosmic rows {mutation.shape[0]}') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7bfdc7d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rows mapped by ENST_name 1724388\n",
      "         ensembl_enst  counts\n",
      "3692  ENST00000529639       2\n",
      "rows mapped by ENSP_name 0\n",
      "Empty DataFrame\n",
      "Columns: [ensembl_ensp, counts]\n",
      "Index: []\n",
      "cosmic without duplicates 1724388\n",
      "cosmic proteins 4018\n"
     ]
    }
   ],
   "source": [
    "# Filter only mutations in our LLPS dataset \n",
    "\n",
    "mutation['internalid'] = range(1, len(mutation)+1)\n",
    "\n",
    "#merge by order 'ensembl_enst', 'ensembl_ensp'\n",
    "#to avoid the rows duplication\n",
    "protein_enst = mutation.merge(protein_enst)\n",
    "mapped_internal = protein_enst['internalid'].unique().tolist()\n",
    "mutation = mutation[~mutation['internalid'].isin(mapped_internal)]\n",
    "print(f'rows mapped by ENST_name {protein_enst.shape[0]}')\n",
    "\n",
    "#analizar enst con disitino id_protein\n",
    "y =  protein_enst[['ensembl_enst', 'id_protein']]\n",
    "y = y.drop_duplicates()\n",
    "dd = y.groupby(['ensembl_enst']).size().reset_index(name='counts')\n",
    "print(dd[dd.counts > 1])\n",
    "#este es un caso raro ya que la secuencia del transcripto es la union de dos uniprots\n",
    "#P35544 + P62861\n",
    "#id_protein 4340 y 4342\n",
    "\n",
    "protein_ensp = mutation.merge(protein_ensp)\n",
    "mapped_internal = protein_ensp['internalid'].unique().tolist()\n",
    "mutation = mutation[~mutation['internalid'].isin(mapped_internal)]\n",
    "print(f'rows mapped by ENSP_name {protein_ensp.shape[0]}')\n",
    "#analisar ensp con disitino id_protein\n",
    "y =  protein_ensp[['ensembl_ensp', 'id_protein']]\n",
    "y = y.drop_duplicates()\n",
    "dd = y.groupby(['ensembl_ensp']).size().reset_index(name='counts')\n",
    "print(dd[dd.counts > 1])\n",
    "\n",
    "#mapear por gene_name, gene_id, hgnc_id\n",
    "\n",
    "\n",
    "mutation = pd.concat([protein_enst, protein_ensp], ignore_index=True)\n",
    "mutation = mutation.drop_duplicates()\n",
    "print(f'cosmic without duplicates {mutation.shape[0]}') \n",
    "print(f'cosmic proteins {len(mutation[\"id_protein\"].unique())}')                \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0e612ece",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['p.V76=' 'p.K102del' 'p.P91S' 'p.E37*' 'p.G78C' 'p.P36S' 'p.P129L'\n",
      " 'p.G128D' 'p.Q96=' 'p.T53I' 'p.A61V' 'p.K127M' 'p.V21G' 'p.Y112C'\n",
      " 'p.R87T' 'p.A26V' 'p.K25N' 'p.E37D' 'p.R82H' 'p.E97=' 'p.P129H' 'p.Y112H'\n",
      " 'p.E9V' 'p.K102T' 'p.A26G' 'p.L3=' 'p.R109Q' 'p.A131T' 'p.R109W' 'p.P36L'\n",
      " 'p.S133C' 'p.A29=' 'p.P36T' 'p.R108W' 'p.R6P' 'p.R6=' 'p.R114W' 'p.A81='\n",
      " 'p.H77R' 'p.K99N' 'p.H27D' 'p.Q111*' 'p.K101=' 'p.K98=' 'p.V86='\n",
      " 'p.K100=' 'p.A61T' 'p.N118=' 'p.G58E' 'p.N132Y' 'p.K92E' 'p.R115H'\n",
      " 'p.E66K']\n"
     ]
    }
   ],
   "source": [
    "print(protein_enst[protein_enst['ensembl_enst'] == 'ENST00000529639']['AA'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ca383815",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1144756\n",
      "        LEGACY_MUTATION_ID  counts\n",
      "8381           COSM1049194       2\n",
      "8382           COSM1049196       2\n",
      "33704          COSM1206683       2\n",
      "59817          COSM1355890       2\n",
      "85748           COSM160375       2\n",
      "...                    ...     ...\n",
      "1063915        COSM9462306       2\n",
      "1071035        COSM9508187       2\n",
      "1084998        COSM9589314       2\n",
      "1090340        COSM9621560       2\n",
      "1099092         COSM967422       2\n",
      "\n",
      "[81 rows x 2 columns]\n",
      "Empty DataFrame\n",
      "Columns: [LEGACY_MUTATION_ID, id_protein, counts]\n",
      "Index: []\n"
     ]
    }
   ],
   "source": [
    "#ver si hay mutiples transcriptos de una misma proteina para eliminarlos \n",
    "#un legacy_mutation_id tiene todas los transcriptos mutados que corresponde a una misma posicion\n",
    "y = mutation[['LEGACY_MUTATION_ID', 'id_protein']]\n",
    "y = y.drop_duplicates()\n",
    "print(y.shape[0])\n",
    "mut_trans = y.groupby(['LEGACY_MUTATION_ID']).size().reset_index(name='counts')\n",
    "mut_trans = mut_trans[mut_trans.counts > 1]\n",
    "print(mut_trans)\n",
    "#puede darse el caso que tengan varios protein_id ya que los uniprot asociados son disitntas variantes de splicing\n",
    "y = mutation[['LEGACY_MUTATION_ID', 'id_protein', 'ensembl_enst']]\n",
    "y = y.drop_duplicates()\n",
    "mut_trans = y.groupby(['LEGACY_MUTATION_ID', 'id_protein']).size().reset_index(name='counts')\n",
    "mut_trans = mut_trans[mut_trans.counts > 1]\n",
    "print(mut_trans)\n",
    "#only one transcript per protein id\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6456a9ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample-mutation 1723613\n",
      "mutation-idprot-transcr 1144756\n"
     ]
    }
   ],
   "source": [
    "#ver si id_mutation and id_protein es unique de la tabla\n",
    "#de ser asi, dejar solo id_mutation y datos de las mutaciones juntas\n",
    "#y otra tabla con \"ID_sample\", \"GENOMIC_MUTATION_ID\", \"LEGACY_MUTATION_ID\", \"MUTATION_ID\", \"Mutation somatic status\", \"Pubmed_PMID\"\n",
    "\n",
    "samples_mut = mutation[[\"id_sample\", \"LEGACY_MUTATION_ID\", 'id_protein', \"somatic_status\", \"pubmed\"]]\n",
    "samples_mut = samples_mut.drop_duplicates()\n",
    "print(f'sample-mutation {samples_mut.shape[0]}')\n",
    "#samples_mut.to_csv('samples_mut.tsv', sep=\"\\t\", index= False)\n",
    "#\"LEGACY_MUTATION_ID\" + 'id_protein' is the mutation id\n",
    "mutation = mutation[[\"LEGACY_MUTATION_ID\", 'id_protein', 'CDS', 'AA', 'consequence', 'genomic']]\n",
    "mutation = mutation.drop_duplicates()\n",
    "print(f'mutation-idprot-transcr {mutation.shape[0]}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0ca7b3e1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['10', '2', '15', '11', '19', '7', '17', '1', '3', '12', '14', '5', '9', '4', '16', '6', '22', '21', '18', '8', '20', '23', '13', '24', '25']\n"
     ]
    }
   ],
   "source": [
    "#split genomic by (x:y-z) in chromosome, start_genomic end_genomic\n",
    "mutation['genomic'] = mutation['genomic'].str.findall('^(.+):(.+)-(.+)$').str[0]\n",
    "mutation['chromosome'] = mutation['genomic'].map(lambda x: x[0])\n",
    "mutation['start_genomic'] = mutation['genomic'].map(lambda x: x[1])\n",
    "mutation['end_genomic'] = mutation['genomic'].map(lambda x: x[2])\n",
    "mutation = mutation.drop(columns=['genomic'])\n",
    "print(mutation['chromosome'].unique().tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "424c7338",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1144756\n",
      "1144756\n",
      "        LEGACY_MUTATION_ID  id_protein  \\\n",
      "0               COSM166059        1011   \n",
      "2              COSM6982134        1011   \n",
      "3               COSM428012        1011   \n",
      "5               COSM427996        1011   \n",
      "6              COSM5703736        1011   \n",
      "...                    ...         ...   \n",
      "1724382        COSM7772448        1844   \n",
      "1724383        COSM4622984        1844   \n",
      "1724384        COSM9330544        1844   \n",
      "1724386        COSM8832530        1844   \n",
      "1724387        COSM3944131        1844   \n",
      "\n",
      "                                                       CDS  \\\n",
      "0                                                  1224dup   \n",
      "2        661_662delinsCCTGCCTGTTGTGAGCTGCTCTACGTGCCCTAC...   \n",
      "3                                             1302_1303dup   \n",
      "5                                                  1203dup   \n",
      "6                                                   633del   \n",
      "...                                                    ...   \n",
      "1724382                                             134C>T   \n",
      "1724383                                             366T>C   \n",
      "1724384                                             124T>C   \n",
      "1724386                                              82G>T   \n",
      "1724387                                               5C>G   \n",
      "\n",
      "                              AA            consequence_cosmic chromosome  \\\n",
      "0                     P409Afs*99        Insertion - Frameshift         10   \n",
      "2        T221delinsPACCELLYVPYVL                           NaN         10   \n",
      "3                     H435Pfs*42        Insertion - Frameshift         10   \n",
      "5                    S402Vfs*106        Insertion - Frameshift         10   \n",
      "6                     S212Pfs*54         Deletion - Frameshift         10   \n",
      "...                          ...                           ...        ...   \n",
      "1724382                     P45L       Substitution - Missense          1   \n",
      "1724383                    D122=  Substitution - coding silent          1   \n",
      "1724384                     F42L       Substitution - Missense          1   \n",
      "1724386                     D28Y       Substitution - Missense          1   \n",
      "1724387                      S2C       Substitution - Missense          1   \n",
      "\n",
      "        start_genomic end_genomic  indexinternal  \n",
      "0             8073912     8073913              1  \n",
      "2             8058724     8058725              2  \n",
      "3             8073991     8073992              3  \n",
      "5             8073891     8073892              4  \n",
      "6             8058696     8058696              5  \n",
      "...               ...         ...            ...  \n",
      "1724382      45515780    45515780        1144752  \n",
      "1724383      45514890    45514890        1144753  \n",
      "1724384      45515790    45515790        1144754  \n",
      "1724386      45518962    45518962        1144755  \n",
      "1724387      45519039    45519039        1144756  \n",
      "\n",
      "[1144756 rows x 9 columns]\n"
     ]
    }
   ],
   "source": [
    "print(mutation.shape[0])\n",
    "mutation = mutation.rename(columns={'consequence': 'consequence_cosmic'})\n",
    "mutation['indexinternal'] = range(1, len(mutation)+1)\n",
    "mutation['AA'] = mutation.AA.map(lambda x: re.findall('^p\\.(.*)$', x))\n",
    "mutation['AA'] = mutation.AA.str[0]\n",
    "mutation = mutation[mutation.AA.notnull()]\n",
    "\n",
    "mutation['CDS'] = mutation.CDS.map(lambda x: re.findall('^c\\.(.*)$', x))\n",
    "mutation['CDS'] = mutation.CDS.str[0]\n",
    "mutation = mutation[mutation.CDS.notnull()]\n",
    "\n",
    "print(mutation.shape[0])\n",
    "print(mutation)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cad313a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def separar_en_cols(df, column, conseq, conseq_regex, override=False):\n",
    "    '''\n",
    "    recibe un DataFrame, el nombre de una columna auxiliar (column)\n",
    "    y un string con el tipo de consecuencia (conseq). La col. auxiliar\n",
    "    es una tupla con los elementos implicados en una mutacion\n",
    "    como la siguiente (aa1, start_pos, aa2, end_pos, aa/s_nuevos).\n",
    "    Devuelve el DataFrame df con estas 5 nuevas columnas\n",
    "    '''\n",
    "    \n",
    "    df_crop = df.copy()\n",
    "      \n",
    "      \n",
    "    if override:\n",
    "        df_crop['aux'] = df_crop[column].str.findall(conseq_regex).str[0]\n",
    "    else:\n",
    "        df_crop['aux'] = df_crop[column].str.findall('^([A-Z\\*])(\\d+)_?([A-Z\\*])?(\\d+)?'+conseq_regex+'(.*)$').str[0]\n",
    "  \n",
    "    df_crop = df_crop[~df_crop['aux'].isnull()]\n",
    "    '''\n",
    "    #check list type bad format    \n",
    "    if conseq in ['synonym', 'nonsense']: #(w, d, '', '', '')\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if ((x[0] == '') or (x[1] == '') or (x[2] != '') or (x[3] != '') or (x[4] != '')) else False)\n",
    "    elif conseq in ['delins']: #(w, d, '', '', w) or (w, d, w, d, w)\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if (((x[0] == '') or (x[1] == '') or (x[4] == '')) or ((x[2] != '') and (x[3] == '')) or ((x[2] == '') and (x[3] != ''))) else False) \n",
    "    elif conseq in ['deletion', 'duplication']: #(w, d, '', '', '') or (w, d, w, d, '')\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if (((x[0] == '') or (x[1] == '') or (x[4] != '')) or ((x[2] != '') and (x[3] == '')) or ((x[2] == '') and (x[3] != ''))) else False) \n",
    "    elif conseq in ['insertion']: #(w, d, w, d, w)\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if ((x[0] == '') or (x[1] == '') or (x[2] == '') or (x[3] == '') or (x[4] == '')) else False) \n",
    "    elif conseq in [\"frameshift\", 'nonstop']: # (w, d, w, '', w) or (w, d, w, '', '')\n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if ((x[0] == '') or (x[1] == '') or (x[2] == '') or (x[3] != '')) else False) \n",
    "        #bad format in nonstop for the delext* (w, d, '', '', w) or (w, d, '', '','')\n",
    "    elif conseq in ['missense']: # (w, d, w, '', '') \n",
    "        df_crop['badformat'] = df_crop['aux'].map(lambda x: True if ((x[0] == '') or (x[1] == '') or (x[2] == '') or (x[3] != '') or (x[4] != '')) else False)  \n",
    "    '''\n",
    "    # start position\n",
    "    df_crop['start_aa'] = df_crop['aux'].map(lambda x: x[1])\n",
    "    # end position\n",
    "    df_crop['end_aa'] = df_crop['aux'].map(lambda x: x[3] if x[3] != '' else x[1]) # poner en el end el start si no hay end\n",
    "        \n",
    "    # from: es el/los aa que cambian # concateno si existe mas de un aa que cambia (o sea, si es un rango)\n",
    "    df_crop['from_aa'] = df_crop['aux'].map(lambda x: str(x[0]) + ('-' + str(x[2]) if x[2] != '' and x[3] != '' else '')) \n",
    "        \n",
    "    if conseq == \"missense\":     \n",
    "        df_crop['to_aa'] = df_crop['aux'].map(lambda x: x[2])\n",
    "    elif conseq == \"nonsense\":\n",
    "        df_crop['to_aa'] = \"*\"\n",
    "    elif conseq in [\"frameshift\", 'nostop']:    \n",
    "        df_crop['to_aa'] = \"\"\n",
    "    else: # to: aa/s nuevos\n",
    "        df_crop['to_aa'] = df_crop['aux'].map(lambda x: x[4])\n",
    "\n",
    "    # consecuencia de la mutacion\n",
    "    df_crop['consequence_protein'] = conseq\n",
    "    #print(df_crop[df_crop['badformat'] == True].sort_values(by='AA', key=lambda col: col.str.len(), ascending=False, na_position='last'))\n",
    "    \n",
    "    df_crop = df_crop.drop(columns=['aux'])\n",
    "\n",
    "    return df_crop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0246f498",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mapping AA mutations\n",
      "all mutations 1144756\n",
      "Found 264184 synonym\n",
      "Found 1606 delins\n",
      "Found 5403 deletions\n",
      "Found 884 inserions\n",
      "Found 38326 frameshift\n",
      "Found 1031 duplications\n",
      "Found 743 nostop\n",
      "Found 60124 nonsense\n",
      "Found 771186 missense\n",
      "mapped 1143487 mutations \n",
      "No mapped 1269 mutations\n"
     ]
    }
   ],
   "source": [
    "# A DataFrame for each molecular consequence\n",
    "print(\"Mapping AA mutations\")\n",
    "mutation_used = mutation.copy()\n",
    "print(f\"all mutations {mutation.shape[0]}\")\n",
    "\n",
    "synonym = separar_en_cols(mutation_used, \"AA\", \"synonym\", \"=$\")\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(synonym['indexinternal'].tolist())]\n",
    "print(f\"Found {synonym.shape[0]} synonym\")\n",
    "\n",
    "\n",
    "delins = separar_en_cols(mutation_used, \"AA\", \"delins\", \"delins\")\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(delins['indexinternal'].tolist())]\n",
    "print(f\"Found {delins.shape[0]} delins\")\n",
    "\n",
    "deletions = separar_en_cols(mutation_used, \"AA\", \"deletion\", \"del$\") # finish with 'del'\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(deletions['indexinternal'].tolist())]\n",
    "print(f\"Found {deletions.shape[0]} deletions\")\n",
    "\n",
    "insertions = separar_en_cols(mutation_used, \"AA\", \"insertion\", \"ins\")\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(insertions['indexinternal'].tolist())]\n",
    "print(f\"Found {insertions.shape[0]} inserions\")\n",
    "\n",
    "frameshift = separar_en_cols(mutation_used, \"AA\", \"frameshift\", 'fs\\*')\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(frameshift['indexinternal'].tolist())]\n",
    "print(f\"Found {frameshift.shape[0]} frameshift\")\n",
    "\n",
    "duplications = separar_en_cols(mutation_used, \"AA\", \"duplication\", \"dup\")\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(duplications['indexinternal'].tolist())]\n",
    "print(f\"Found {duplications.shape[0]} duplications\")\n",
    "\n",
    "nostop = separar_en_cols(mutation_used, \"AA\", \"nonstop\", '(?:del)?ext\\*')\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(nostop['indexinternal'].tolist())]\n",
    "print(f\"Found {nostop.shape[0]} nostop\")\n",
    "\n",
    "nonsense = separar_en_cols(mutation_used, \"AA\", \"nonsense\", \"\\*$\")\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(nonsense['indexinternal'].tolist())]\n",
    "print(f\"Found {nonsense.shape[0]} nonsense\")\n",
    "\n",
    "missense = separar_en_cols(mutation_used, \"AA\", \"missense\", '^([A-Z])(\\d+)([A-Z])$(.*)?(.*)?$', override=True)\n",
    "print(f\"Found {missense.shape[0]} missense\")\n",
    "\n",
    "#los frameshift M1?, M1_?\\d*\n",
    "## Concatenate everything\n",
    "mutation_mapped = pd.concat([synonym, delins, deletions, insertions, frameshift, duplications, nostop, nonsense, missense])\n",
    "\n",
    "print(f\"mapped {mutation_mapped.shape[0]} mutations \")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(mutation_mapped['indexinternal'].tolist())]\n",
    "print(f\"No mapped {mutation_used.shape[0]} mutations\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e486245f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def separar_en_cols_cds(df, column, conseq, conseq_regex, override=False):\n",
    "    '''\n",
    "    recibe un DataFrame, el nombre de una columna auxiliar (column)\n",
    "    y un string con el tipo de consecuencia (conseq). La col. auxiliar\n",
    "    es una tupla con los elementos implicados en una mutacion\n",
    "    como la siguiente (nucleatides1, nucleatides2).\n",
    "    Devuelve el DataFrame df con estas 2 nuevas columnas\n",
    "    '''\n",
    "    \n",
    "    df_crop = df.copy()\n",
    "      \n",
    "      \n",
    "    if override:\n",
    "        df_crop['aux'] = df_crop[column].str.findall(conseq_regex).str[0]\n",
    "    else:\n",
    "        df_crop['aux'] = df_crop[column].str.findall('^[-\\+\\d]+_?(?:[-\\+\\d]+)?'+conseq_regex+'([ACTG]+)?>?([ACTG]+)?$').str[0]\n",
    "  \n",
    "    df_crop = df_crop[~df_crop['aux'].isnull()]\n",
    "    \n",
    "    df_crop['from_genomic'] = ''\n",
    "    df_crop['to_genomic'] = ''\n",
    "    \n",
    "    if conseq in [\"delins\", 'insertion']: \n",
    "        df_crop['to_genomic'] = df_crop['aux'].map(lambda x: x[0])\n",
    "    elif conseq == 'missense': \n",
    "        df_crop['from_genomic'] = df_crop['aux'].map(lambda x: x[0])\n",
    "        df_crop['to_genomic'] = df_crop['aux'].map(lambda x: x[1])\n",
    "\n",
    "    # consecuencia de la mutacion\n",
    "    df_crop['consequence_genomic'] = conseq\n",
    "    #print(df_crop[df_crop['badformat'] == True].sort_values(by='AA', key=lambda col: col.str.len(), ascending=False, na_position='last'))\n",
    "    \n",
    "    df_crop = df_crop.drop(columns=['aux'])\n",
    "\n",
    "    return df_crop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3fc48fba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mapping CDS mutations\n",
      "all mutations 1143487\n",
      "Found 9032 delins cds\n",
      "Found 31123 deletions cds\n",
      "Found 4077 inserions cds\n",
      "Found 12871 duplications cds\n",
      "Found 317 inversion cds\n",
      "Found 771186 missense cds\n",
      "mapped 1143487 mutations \n",
      "No mapped 0 mutations\n"
     ]
    }
   ],
   "source": [
    "print(\"Mapping CDS mutations\")\n",
    "\n",
    "mutation_used = mutation_mapped.copy()\n",
    "print(f\"all mutations {mutation_used.shape[0]}\")\n",
    "\n",
    "delins = separar_en_cols_cds(mutation_used, \"CDS\", \"delins\", \"delins\")\n",
    "print(f\"Found {delins.shape[0]} delins cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(delins['indexinternal'].tolist())]\n",
    "deletions = separar_en_cols_cds(mutation_used, \"CDS\", \"deletion\", \"del$\") # finish with 'del'\n",
    "print(f\"Found {deletions.shape[0]} deletions cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(deletions['indexinternal'].tolist())]\n",
    "insertions = separar_en_cols_cds(mutation_used, \"CDS\", \"insertion\", \"ins\")\n",
    "print(f\"Found {insertions.shape[0]} inserions cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(insertions['indexinternal'].tolist())]\n",
    "duplications = separar_en_cols_cds(mutation_used, \"CDS\", \"duplication\", \"dup$\")\n",
    "print(f\"Found {duplications.shape[0]} duplications cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(duplications['indexinternal'].tolist())]\n",
    "inversions = separar_en_cols_cds(mutation_used, \"CDS\", \"invertion\", \"inv$\")\n",
    "print(f\"Found {inversions.shape[0]} inversion cds\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(inversions['indexinternal'].tolist())]\n",
    "substitution = separar_en_cols_cds(mutation_used, \"CDS\", \"missense\", '')\n",
    "print(f\"Found {missense.shape[0]} missense cds\")\n",
    "\n",
    "mutation_mapped2 = pd.concat([delins, deletions, insertions, duplications, inversions, substitution])\n",
    "\n",
    "print(f\"mapped {mutation_mapped2.shape[0]} mutations \")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(mutation_mapped2['indexinternal'].tolist())]\n",
    "print(f\"No mapped {mutation_used.shape[0]} mutations\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "d2cc4df2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1143487\n",
      "1143300\n",
      "1119621\n",
      "        start_aa end_aa to_aa from_aa  id_protein  counts\n",
      "394797       241    241             S         623       4\n",
      "69287       1157   1157             E        1020       4\n",
      "413360       251    251             I         623       4\n",
      "415060       252    252             L         623       4\n",
      "416902       253    253             T         623       4\n",
      "...          ...    ...   ...     ...         ...     ...\n",
      "355178       221    221             P         242       2\n",
      "355190       221    221             P        2201       2\n",
      "356684      2216   2216             P         507       2\n",
      "356685      2216   2216             Q        3004       2\n",
      "1114290      997    997             Y        1020       2\n",
      "\n",
      "[4392 rows x 6 columns]\n",
      "1115104\n"
     ]
    }
   ],
   "source": [
    "print(mutation_mapped2.shape[0])\n",
    "a = mutation_mapped2[['start_aa', 'end_aa', 'to_aa', 'from_aa', 'id_protein', 'consequence_protein', 'start_genomic', 'end_genomic', 'from_genomic', 'to_genomic', 'consequence_genomic']].drop_duplicates()\n",
    "print(a.shape[0])\n",
    "\n",
    "\n",
    "\n",
    "x = mutation_mapped2[['start_aa', 'end_aa', 'to_aa', 'from_aa', 'id_protein', 'consequence_protein']].drop_duplicates()\n",
    "print(x.shape[0])\n",
    "\n",
    "y = x.groupby(['start_aa', 'end_aa', 'to_aa', 'from_aa', 'id_protein']).size().reset_index(name='counts')\n",
    "y = y[y.counts > 1]\n",
    "y = y.sort_values(by='counts', ascending=False)\n",
    "print(y)\n",
    "\n",
    "x = mutation_mapped2[['start_aa', 'end_aa', 'to_aa', 'from_aa', 'id_protein']].drop_duplicates()\n",
    "print(x.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "2fd6f182",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1789\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>LEGACY_MUTATION_ID</th>\n",
       "      <th>id_protein</th>\n",
       "      <th>CDS</th>\n",
       "      <th>AA</th>\n",
       "      <th>consequence_cosmic</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_genomic</th>\n",
       "      <th>end_genomic</th>\n",
       "      <th>indexinternal</th>\n",
       "      <th>start_aa</th>\n",
       "      <th>end_aa</th>\n",
       "      <th>from_aa</th>\n",
       "      <th>to_aa</th>\n",
       "      <th>consequence_protein</th>\n",
       "      <th>from_genomic</th>\n",
       "      <th>to_genomic</th>\n",
       "      <th>consequence_genomic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1496511</th>\n",
       "      <td>COSM45017</td>\n",
       "      <td>623</td>\n",
       "      <td>722_723delinsTT</td>\n",
       "      <td>S241F</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>17</td>\n",
       "      <td>7674240</td>\n",
       "      <td>7674241</td>\n",
       "      <td>1039810</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td>F</td>\n",
       "      <td>missense</td>\n",
       "      <td></td>\n",
       "      <td>TT</td>\n",
       "      <td>delins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1497270</th>\n",
       "      <td>COSM45168</td>\n",
       "      <td>623</td>\n",
       "      <td>722_724del</td>\n",
       "      <td>S241del</td>\n",
       "      <td>Deletion - In frame</td>\n",
       "      <td>17</td>\n",
       "      <td>7674240</td>\n",
       "      <td>7674242</td>\n",
       "      <td>1039911</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>deletion</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1500937</th>\n",
       "      <td>COSM5966959</td>\n",
       "      <td>623</td>\n",
       "      <td>720_722del</td>\n",
       "      <td>S241del</td>\n",
       "      <td>Deletion - In frame</td>\n",
       "      <td>17</td>\n",
       "      <td>7674241</td>\n",
       "      <td>7674243</td>\n",
       "      <td>1040312</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>deletion</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1494914</th>\n",
       "      <td>COSM43645</td>\n",
       "      <td>623</td>\n",
       "      <td>721del</td>\n",
       "      <td>S241Pfs*6</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>17</td>\n",
       "      <td>7674243</td>\n",
       "      <td>7674243</td>\n",
       "      <td>1039602</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>frameshift</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1499986</th>\n",
       "      <td>COSM9345425</td>\n",
       "      <td>623</td>\n",
       "      <td>722_732del</td>\n",
       "      <td>S241Wfs*19</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>17</td>\n",
       "      <td>7674231</td>\n",
       "      <td>7674241</td>\n",
       "      <td>1040220</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>frameshift</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519450</th>\n",
       "      <td>COSM45250</td>\n",
       "      <td>623</td>\n",
       "      <td>720_721del</td>\n",
       "      <td>S241Lfs*22</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>17</td>\n",
       "      <td>7674242</td>\n",
       "      <td>7674243</td>\n",
       "      <td>1041742</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>frameshift</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1522620</th>\n",
       "      <td>COSM45788</td>\n",
       "      <td>623</td>\n",
       "      <td>722_723del</td>\n",
       "      <td>S241Lfs*22</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>17</td>\n",
       "      <td>7674240</td>\n",
       "      <td>7674241</td>\n",
       "      <td>1041936</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>frameshift</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1526895</th>\n",
       "      <td>COSM9306816</td>\n",
       "      <td>623</td>\n",
       "      <td>721_722del</td>\n",
       "      <td>S241Lfs*22</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>17</td>\n",
       "      <td>7674241</td>\n",
       "      <td>7674242</td>\n",
       "      <td>1042184</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>frameshift</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1494313</th>\n",
       "      <td>COSM6929598</td>\n",
       "      <td>623</td>\n",
       "      <td>722_724dup</td>\n",
       "      <td>S241dup</td>\n",
       "      <td>Insertion - In frame</td>\n",
       "      <td>17</td>\n",
       "      <td>7674242</td>\n",
       "      <td>7674243</td>\n",
       "      <td>1039516</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>duplication</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>duplication</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1500556</th>\n",
       "      <td>COSM46004</td>\n",
       "      <td>623</td>\n",
       "      <td>723C&gt;T</td>\n",
       "      <td>S241=</td>\n",
       "      <td>Substitution - coding silent</td>\n",
       "      <td>17</td>\n",
       "      <td>7674240</td>\n",
       "      <td>7674240</td>\n",
       "      <td>1040282</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>synonym</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1506084</th>\n",
       "      <td>COSM43824</td>\n",
       "      <td>623</td>\n",
       "      <td>723C&gt;A</td>\n",
       "      <td>S241=</td>\n",
       "      <td>Substitution - coding silent</td>\n",
       "      <td>17</td>\n",
       "      <td>7674240</td>\n",
       "      <td>7674240</td>\n",
       "      <td>1040800</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>synonym</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1514279</th>\n",
       "      <td>COSM44158</td>\n",
       "      <td>623</td>\n",
       "      <td>723C&gt;G</td>\n",
       "      <td>S241=</td>\n",
       "      <td>Substitution - coding silent</td>\n",
       "      <td>17</td>\n",
       "      <td>7674240</td>\n",
       "      <td>7674240</td>\n",
       "      <td>1041440</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td></td>\n",
       "      <td>synonym</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1490383</th>\n",
       "      <td>COSM10812</td>\n",
       "      <td>623</td>\n",
       "      <td>722C&gt;T</td>\n",
       "      <td>S241F</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>17</td>\n",
       "      <td>7674241</td>\n",
       "      <td>7674241</td>\n",
       "      <td>1038459</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td>F</td>\n",
       "      <td>missense</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1491478</th>\n",
       "      <td>COSM10709</td>\n",
       "      <td>623</td>\n",
       "      <td>722C&gt;G</td>\n",
       "      <td>S241C</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>17</td>\n",
       "      <td>7674241</td>\n",
       "      <td>7674241</td>\n",
       "      <td>1038903</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td>C</td>\n",
       "      <td>missense</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1491555</th>\n",
       "      <td>COSM10935</td>\n",
       "      <td>623</td>\n",
       "      <td>722C&gt;A</td>\n",
       "      <td>S241Y</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>17</td>\n",
       "      <td>7674241</td>\n",
       "      <td>7674241</td>\n",
       "      <td>1038930</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td>Y</td>\n",
       "      <td>missense</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1493425</th>\n",
       "      <td>COSM44578</td>\n",
       "      <td>623</td>\n",
       "      <td>721T&gt;C</td>\n",
       "      <td>S241P</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>17</td>\n",
       "      <td>7674242</td>\n",
       "      <td>7674242</td>\n",
       "      <td>1039342</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td>P</td>\n",
       "      <td>missense</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1493714</th>\n",
       "      <td>COSM44067</td>\n",
       "      <td>623</td>\n",
       "      <td>721T&gt;A</td>\n",
       "      <td>S241T</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>17</td>\n",
       "      <td>7674242</td>\n",
       "      <td>7674242</td>\n",
       "      <td>1039405</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td>T</td>\n",
       "      <td>missense</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1496625</th>\n",
       "      <td>COSM44224</td>\n",
       "      <td>623</td>\n",
       "      <td>721T&gt;G</td>\n",
       "      <td>S241A</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>17</td>\n",
       "      <td>7674242</td>\n",
       "      <td>7674242</td>\n",
       "      <td>1039830</td>\n",
       "      <td>241</td>\n",
       "      <td>241</td>\n",
       "      <td>S</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        LEGACY_MUTATION_ID  id_protein              CDS          AA  \\\n",
       "1496511          COSM45017         623  722_723delinsTT       S241F   \n",
       "1497270          COSM45168         623       722_724del     S241del   \n",
       "1500937        COSM5966959         623       720_722del     S241del   \n",
       "1494914          COSM43645         623           721del   S241Pfs*6   \n",
       "1499986        COSM9345425         623       722_732del  S241Wfs*19   \n",
       "1519450          COSM45250         623       720_721del  S241Lfs*22   \n",
       "1522620          COSM45788         623       722_723del  S241Lfs*22   \n",
       "1526895        COSM9306816         623       721_722del  S241Lfs*22   \n",
       "1494313        COSM6929598         623       722_724dup     S241dup   \n",
       "1500556          COSM46004         623           723C>T       S241=   \n",
       "1506084          COSM43824         623           723C>A       S241=   \n",
       "1514279          COSM44158         623           723C>G       S241=   \n",
       "1490383          COSM10812         623           722C>T       S241F   \n",
       "1491478          COSM10709         623           722C>G       S241C   \n",
       "1491555          COSM10935         623           722C>A       S241Y   \n",
       "1493425          COSM44578         623           721T>C       S241P   \n",
       "1493714          COSM44067         623           721T>A       S241T   \n",
       "1496625          COSM44224         623           721T>G       S241A   \n",
       "\n",
       "                   consequence_cosmic chromosome start_genomic end_genomic  \\\n",
       "1496511       Substitution - Missense         17       7674240     7674241   \n",
       "1497270           Deletion - In frame         17       7674240     7674242   \n",
       "1500937           Deletion - In frame         17       7674241     7674243   \n",
       "1494914         Deletion - Frameshift         17       7674243     7674243   \n",
       "1499986         Deletion - Frameshift         17       7674231     7674241   \n",
       "1519450         Deletion - Frameshift         17       7674242     7674243   \n",
       "1522620         Deletion - Frameshift         17       7674240     7674241   \n",
       "1526895         Deletion - Frameshift         17       7674241     7674242   \n",
       "1494313          Insertion - In frame         17       7674242     7674243   \n",
       "1500556  Substitution - coding silent         17       7674240     7674240   \n",
       "1506084  Substitution - coding silent         17       7674240     7674240   \n",
       "1514279  Substitution - coding silent         17       7674240     7674240   \n",
       "1490383       Substitution - Missense         17       7674241     7674241   \n",
       "1491478       Substitution - Missense         17       7674241     7674241   \n",
       "1491555       Substitution - Missense         17       7674241     7674241   \n",
       "1493425       Substitution - Missense         17       7674242     7674242   \n",
       "1493714       Substitution - Missense         17       7674242     7674242   \n",
       "1496625       Substitution - Missense         17       7674242     7674242   \n",
       "\n",
       "         indexinternal start_aa end_aa from_aa to_aa consequence_protein  \\\n",
       "1496511        1039810      241    241       S     F            missense   \n",
       "1497270        1039911      241    241       S                  deletion   \n",
       "1500937        1040312      241    241       S                  deletion   \n",
       "1494914        1039602      241    241       S                frameshift   \n",
       "1499986        1040220      241    241       S                frameshift   \n",
       "1519450        1041742      241    241       S                frameshift   \n",
       "1522620        1041936      241    241       S                frameshift   \n",
       "1526895        1042184      241    241       S                frameshift   \n",
       "1494313        1039516      241    241       S               duplication   \n",
       "1500556        1040282      241    241       S                   synonym   \n",
       "1506084        1040800      241    241       S                   synonym   \n",
       "1514279        1041440      241    241       S                   synonym   \n",
       "1490383        1038459      241    241       S     F            missense   \n",
       "1491478        1038903      241    241       S     C            missense   \n",
       "1491555        1038930      241    241       S     Y            missense   \n",
       "1493425        1039342      241    241       S     P            missense   \n",
       "1493714        1039405      241    241       S     T            missense   \n",
       "1496625        1039830      241    241       S     A            missense   \n",
       "\n",
       "        from_genomic to_genomic consequence_genomic  \n",
       "1496511                      TT              delins  \n",
       "1497270                                    deletion  \n",
       "1500937                                    deletion  \n",
       "1494914                                    deletion  \n",
       "1499986                                    deletion  \n",
       "1519450                                    deletion  \n",
       "1522620                                    deletion  \n",
       "1526895                                    deletion  \n",
       "1494313                                 duplication  \n",
       "1500556            C          T            missense  \n",
       "1506084            C          A            missense  \n",
       "1514279            C          G            missense  \n",
       "1490383            C          T            missense  \n",
       "1491478            C          G            missense  \n",
       "1491555            C          A            missense  \n",
       "1493425            T          C            missense  \n",
       "1493714            T          A            missense  \n",
       "1496625            T          G            missense  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(y['id_protein'].unique()))\n",
    "mutation_mapped2[((mutation_mapped2['id_protein'] == 623) & (mutation_mapped2['start_aa'] == '241') & (mutation_mapped2['end_aa'] == '241') & (mutation_mapped2['from_aa'] == 'S'))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c5d6653",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
