{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5774d842",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "\n",
    "p_prot = pd.read_csv(\"../db_tables/protein.tsv\", sep=\"\\t\")\n",
    "p_prot = p_prot[['id_protein', 'uniprot_acc']]\n",
    "\n",
    "#by ensemble transcript\n",
    "protein_enst = pd.read_csv(\"../raw_data/llps_uniprot2ENST.tab.txt\", sep=\"\\t\")\n",
    "protein_enst = protein_enst.rename(columns={'From': 'uniprot_acc', 'To':'ensembl_enst'})\n",
    "protein_enst = protein_enst.merge(p_prot[['id_protein', 'uniprot_acc']]).drop(columns=['uniprot_acc'])\n",
    "\n",
    "#by ensemble protein\n",
    "protein_ensp = pd.read_csv(\"../raw_data/llps_uniprot2ENSP.tab.txt\", sep=\"\\t\")\n",
    "protein_ensp = protein_ensp.rename(columns={'From': 'uniprot_acc', 'To':'ensembl_ensp'})\n",
    "protein_ensp = protein_ensp.merge(p_prot[['id_protein', 'uniprot_acc']]).drop(columns=['uniprot_acc'])\n",
    "\n",
    "datafile  = \"../raw_data/cosmic_v94/CosmicMutantExport_sel.tsv\"\n",
    "#for each sample_id has the cancer types columns and the cosmic_phenotype_id\n",
    "samples_data = \"../raw_data/cosmic_v94/CosmicSample.tsv\" \n",
    "#for each cosmic_phenotype_id has the NCI_CODE and other like EFO, MONDO, ....\n",
    "class_data = \"../raw_data/cosmic_v94/classification.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6339c492",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['id_sample', 'GENOMIC_MUTATION_ID', 'LEGACY_MUTATION_ID', 'MUTATION_ID', 'CDS', 'AA', 'consequence', 'genomic', 'somatic_status', 'pubmed', 'ensembl_ensp', 'ensembl_enst']\n",
      "   id_sample GENOMIC_MUTATION_ID LEGACY_MUTATION_ID  MUTATION_ID  \\\n",
      "0    2198328        COSV57322711        COSM1628357     98293492   \n",
      "1    2869867        COSV57321510        COSM1706236     98292376   \n",
      "2    2549254        COSV57321745        COSM5796874     98292486   \n",
      "3    2749203        COSV57324149        COSM7003726     98292523   \n",
      "4    2839416       COSV104562760        COSM9463514     98293844   \n",
      "\n",
      "                 CDS       AA              consequence  \\\n",
      "0           c.609G>T  p.M203I  Substitution - Missense   \n",
      "1           c.781G>A  p.E261K  Substitution - Missense   \n",
      "2           c.385G>C  p.G129R  Substitution - Missense   \n",
      "3            c.20G>C    p.G7A  Substitution - Missense   \n",
      "4  c.559_560delinsAA  p.G187K  Substitution - Missense   \n",
      "\n",
      "                  genomic             somatic_status    pubmed  \\\n",
      "0  12:108894364-108894364  Confirmed somatic variant       NaN   \n",
      "1  12:108898764-108898764  Confirmed somatic variant  26950094   \n",
      "2  12:108889544-108889544  Confirmed somatic variant       NaN   \n",
      "3  12:108885026-108885026  Confirmed somatic variant  29253671   \n",
      "4  12:108894314-108894315  Confirmed somatic variant  28467829   \n",
      "\n",
      "      ensembl_ensp     ensembl_enst  \n",
      "0  ENSP00000228476  ENST00000228476  \n",
      "1  ENSP00000228476  ENST00000228476  \n",
      "2  ENSP00000228476  ENST00000228476  \n",
      "3  ENSP00000228476  ENST00000228476  \n",
      "4  ENSP00000228476  ENST00000228476  \n",
      "cosmic rows 6795791\n"
     ]
    }
   ],
   "source": [
    "mutation = pd.read_csv(datafile, sep=\"\\t\")\n",
    "#to string the columns\n",
    "mutation['pubmed'] = mutation['pubmed'].map(lambda x: str(int(x)) if not np.isnan(x) else x)\n",
    "print(list(mutation.columns))\n",
    "print(mutation.head())\n",
    "print(f'cosmic rows {mutation.shape[0]}') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7bfdc7d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         ensembl_enst  counts\n",
      "3692  ENST00000529639       2\n",
      "Empty DataFrame\n",
      "Columns: [ensembl_ensp, counts]\n",
      "Index: []\n",
      "rows mapped by ENSP_name 0\n",
      "rows mapped by ENST_name 1724388\n",
      "cosmic without duplicates 1724388\n",
      "cosmic proteins 4018\n"
     ]
    }
   ],
   "source": [
    "# Filter only mutations in our LLPS dataset \n",
    "\n",
    "mutation['internalid'] = range(1, len(mutation)+1)\n",
    "\n",
    "#merge by order 'ensembl_enst', 'ensembl_ensp'\n",
    "#to avoid the rows duplication\n",
    "protein_enst = mutation.merge(protein_enst)\n",
    "\n",
    "y =  protein_enst[['ensembl_enst', 'id_protein']]\n",
    "y = y.drop_duplicates()\n",
    "dd = y.groupby(['ensembl_enst']).size().reset_index(name='counts')\n",
    "print(dd[dd.counts > 1])\n",
    "#este es un caso raro ya que la secuencia del transcripto es la union de dos uniprots\n",
    "#P35544 + P62861\n",
    "#id_protein 4340 y 4342\n",
    "\n",
    "mapped_internal = protein_enst['internalid'].unique().tolist()\n",
    "mutation = mutation[~mutation['internalid'].isin(mapped_internal)]\n",
    "protein_ensp = mutation.merge(protein_ensp)\n",
    "y =  protein_ensp[['ensembl_ensp', 'id_protein']]\n",
    "y = y.drop_duplicates()\n",
    "dd = y.groupby(['ensembl_ensp']).size().reset_index(name='counts')\n",
    "print(dd[dd.counts > 1])\n",
    "\n",
    "print(f'rows mapped by ENSP_name {protein_ensp.shape[0]}')\n",
    "print(f'rows mapped by ENST_name {protein_enst.shape[0]}')\n",
    "\n",
    "mutation = pd.concat([protein_enst, protein_ensp], ignore_index=True)\n",
    "mutation = mutation.drop_duplicates()\n",
    "print(f'cosmic without duplicates {mutation.shape[0]}') \n",
    "print(f'cosmic proteins {len(mutation[\"id_protein\"].unique())}')                \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ca383815",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1144756\n",
      "        LEGACY_MUTATION_ID  counts\n",
      "8381           COSM1049194       2\n",
      "8382           COSM1049196       2\n",
      "33704          COSM1206683       2\n",
      "59817          COSM1355890       2\n",
      "85748           COSM160375       2\n",
      "...                    ...     ...\n",
      "1063915        COSM9462306       2\n",
      "1071035        COSM9508187       2\n",
      "1084998        COSM9589314       2\n",
      "1090340        COSM9621560       2\n",
      "1099092         COSM967422       2\n",
      "\n",
      "[81 rows x 2 columns]\n",
      "Empty DataFrame\n",
      "Columns: [LEGACY_MUTATION_ID, id_protein, counts]\n",
      "Index: []\n"
     ]
    }
   ],
   "source": [
    "#ver si hay mutiples transcriptos de una misma proteina para eliminarlos \n",
    "#un legacy_mutation_id tiene todas los transcriptos mutados que corresponde a una misma posicion\n",
    "y = mutation[['LEGACY_MUTATION_ID', 'id_protein']]\n",
    "y = y.drop_duplicates()\n",
    "print(y.shape[0])\n",
    "mut_trans = y.groupby(['LEGACY_MUTATION_ID']).size().reset_index(name='counts')\n",
    "mut_trans = mut_trans[mut_trans.counts > 1]\n",
    "print(mut_trans)\n",
    "#puede darse el caso que tengan varios protein_id ya que los uniprot asociados son disitntas variantes de splicing\n",
    "y = mutation[['LEGACY_MUTATION_ID', 'id_protein', 'ensembl_enst']]\n",
    "y = y.drop_duplicates()\n",
    "mut_trans = y.groupby(['LEGACY_MUTATION_ID', 'id_protein']).size().reset_index(name='counts')\n",
    "mut_trans = mut_trans[mut_trans.counts > 1]\n",
    "print(mut_trans)\n",
    "#only one transcript per protein id\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6456a9ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample-mutation 1723613\n",
      "mutation-idprot-transcr 1144756\n"
     ]
    }
   ],
   "source": [
    "#ver si id_mutation and id_protein es unique de la tabla\n",
    "#de ser asi, dejar solo id_mutation y datos de las mutaciones juntas\n",
    "#y otra tabla con \"ID_sample\", \"GENOMIC_MUTATION_ID\", \"LEGACY_MUTATION_ID\", \"MUTATION_ID\", \"Mutation somatic status\", \"Pubmed_PMID\"\n",
    "\n",
    "samples_mut = mutation[[\"id_sample\", \"LEGACY_MUTATION_ID\", 'id_protein', \"somatic_status\", \"pubmed\"]]\n",
    "samples_mut = samples_mut.drop_duplicates()\n",
    "print(f'sample-mutation {samples_mut.shape[0]}')\n",
    "#samples_mut.to_csv('samples_mut.tsv', sep=\"\\t\", index= False)\n",
    "#\"LEGACY_MUTATION_ID\" + 'id_protein' is the mutation id\n",
    "mutation = mutation[[\"LEGACY_MUTATION_ID\", 'id_protein', 'CDS', 'AA', 'consequence', 'genomic']]\n",
    "mutation = mutation.drop_duplicates()\n",
    "print(f'mutation-idprot-transcr {mutation.shape[0]}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0ca7b3e1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['10', '2', '15', '11', '19', '7', '17', '1', '3', '12', '14', '5', '9', '4', '16', '6', '22', '21', '18', '8', '20', '23', '13', '24', '25']\n"
     ]
    }
   ],
   "source": [
    "#split genomic by (x:y-z) in chromosome, start_genomic end_genomic\n",
    "mutation['genomic'] = mutation['genomic'].str.findall('^(.+):(.+)-(.+)$').str[0]\n",
    "mutation['chromosome'] = mutation['genomic'].map(lambda x: x[0])\n",
    "mutation['start_genomic'] = mutation['genomic'].map(lambda x: x[1])\n",
    "mutation['end_genomic'] = mutation['genomic'].map(lambda x: x[2])\n",
    "mutation = mutation.drop(columns=['genomic'])\n",
    "print(mutation['chromosome'].unique().tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "424c7338",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1144756\n",
      "1144756\n",
      "        LEGACY_MUTATION_ID  id_protein  \\\n",
      "0               COSM166059        1011   \n",
      "2              COSM6982134        1011   \n",
      "3               COSM428012        1011   \n",
      "5               COSM427996        1011   \n",
      "6              COSM5703736        1011   \n",
      "...                    ...         ...   \n",
      "1724382        COSM7772448        1844   \n",
      "1724383        COSM4622984        1844   \n",
      "1724384        COSM9330544        1844   \n",
      "1724386        COSM8832530        1844   \n",
      "1724387        COSM3944131        1844   \n",
      "\n",
      "                                                       CDS  \\\n",
      "0                                                  1224dup   \n",
      "2        661_662delinsCCTGCCTGTTGTGAGCTGCTCTACGTGCCCTAC...   \n",
      "3                                             1302_1303dup   \n",
      "5                                                  1203dup   \n",
      "6                                                   633del   \n",
      "...                                                    ...   \n",
      "1724382                                             134C>T   \n",
      "1724383                                             366T>C   \n",
      "1724384                                             124T>C   \n",
      "1724386                                              82G>T   \n",
      "1724387                                               5C>G   \n",
      "\n",
      "                              AA            consequence_cosmic chromosome  \\\n",
      "0                     P409Afs*99        Insertion - Frameshift         10   \n",
      "2        T221delinsPACCELLYVPYVL                           NaN         10   \n",
      "3                     H435Pfs*42        Insertion - Frameshift         10   \n",
      "5                    S402Vfs*106        Insertion - Frameshift         10   \n",
      "6                     S212Pfs*54         Deletion - Frameshift         10   \n",
      "...                          ...                           ...        ...   \n",
      "1724382                     P45L       Substitution - Missense          1   \n",
      "1724383                    D122=  Substitution - coding silent          1   \n",
      "1724384                     F42L       Substitution - Missense          1   \n",
      "1724386                     D28Y       Substitution - Missense          1   \n",
      "1724387                      S2C       Substitution - Missense          1   \n",
      "\n",
      "        start_genomic end_genomic  indexinternal  \n",
      "0             8073912     8073913              1  \n",
      "2             8058724     8058725              2  \n",
      "3             8073991     8073992              3  \n",
      "5             8073891     8073892              4  \n",
      "6             8058696     8058696              5  \n",
      "...               ...         ...            ...  \n",
      "1724382      45515780    45515780        1144752  \n",
      "1724383      45514890    45514890        1144753  \n",
      "1724384      45515790    45515790        1144754  \n",
      "1724386      45518962    45518962        1144755  \n",
      "1724387      45519039    45519039        1144756  \n",
      "\n",
      "[1144756 rows x 9 columns]\n"
     ]
    }
   ],
   "source": [
    "print(mutation.shape[0])\n",
    "mutation = mutation.rename(columns={'consequence': 'consequence_cosmic'})\n",
    "mutation['indexinternal'] = range(1, len(mutation)+1)\n",
    "mutation['AA'] = mutation.AA.map(lambda x: re.findall('^p\\.(.*)$', x))\n",
    "mutation['AA'] = mutation.AA.str[0]\n",
    "mutation = mutation[mutation.AA.notnull()]\n",
    "\n",
    "mutation['CDS'] = mutation.CDS.map(lambda x: re.findall('^c\\.(.*)$', x))\n",
    "mutation['CDS'] = mutation.CDS.str[0]\n",
    "mutation = mutation[mutation.CDS.notnull()]\n",
    "\n",
    "print(mutation.shape[0])\n",
    "print(mutation)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "57b6fc18",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Insertion - Frameshift', nan, 'Deletion - Frameshift', 'Substitution - coding silent', 'Substitution - Missense', 'Deletion - In frame', 'Substitution - Nonsense', 'Complex - frameshift', 'Insertion - In frame', 'Nonstop extension', 'Complex - deletion inframe', 'Frameshift', 'Complex - insertion inframe']\n",
      "     LEGACY_MUTATION_ID  id_protein  \\\n",
      "2           COSM6982134        1011   \n",
      "1566        COSM4603688        1011   \n",
      "1920        COSM4856816         243   \n",
      "3817        COSM5351343        2188   \n",
      "4192        COSM5816736        1865   \n",
      "\n",
      "                                                    CDS  \\\n",
      "2     661_662delinsCCTGCCTGTTGTGAGCTGCTCTACGTGCCCTAC...   \n",
      "1566                                929_940delinsTCCGAC   \n",
      "1920                                             689G>A   \n",
      "3817                                            1430G>A   \n",
      "4192                                            2139G>A   \n",
      "\n",
      "                           AA consequence_cosmic chromosome start_genomic  \\\n",
      "2     T221delinsPACCELLYVPYVL                NaN         10       8058724   \n",
      "1566       A310_A314delinsVRP                NaN         10       8069477   \n",
      "1920                    *230=                NaN         15      41309755   \n",
      "3817                    *477=                NaN          7     100873302   \n",
      "4192                    *713=                NaN         19      18210929   \n",
      "\n",
      "     end_genomic  indexinternal  \n",
      "2        8058725              2  \n",
      "1566     8069488            748  \n",
      "1920    41309755            956  \n",
      "3817   100873302           2382  \n",
      "4192    18210929           2671  \n",
      "   LEGACY_MUTATION_ID  id_protein           CDS           AA  \\\n",
      "0          COSM166059        1011       1224dup   P409Afs*99   \n",
      "3          COSM428012        1011  1302_1303dup   H435Pfs*42   \n",
      "5          COSM427996        1011       1203dup  S402Vfs*106   \n",
      "7          COSM428007        1011       1277dup   S427Ifs*81   \n",
      "13        COSM5231132        1011  1005_1006dup   D336Gfs*21   \n",
      "\n",
      "        consequence_cosmic chromosome start_genomic end_genomic  indexinternal  \n",
      "0   Insertion - Frameshift         10       8073912     8073913              1  \n",
      "3   Insertion - Frameshift         10       8073991     8073992              3  \n",
      "5   Insertion - Frameshift         10       8073891     8073892              4  \n",
      "7   Insertion - Frameshift         10       8073965     8073966              6  \n",
      "13  Insertion - Frameshift         10       8069554     8069555             12  \n",
      "   LEGACY_MUTATION_ID  id_protein           CDS          AA  \\\n",
      "6         COSM5703736        1011        633del  S212Pfs*54   \n",
      "8         COSM7089781        1011       1278del  S427Pfs*49   \n",
      "14        COSM9180417        1011  1221_1228del  S408Qfs*97   \n",
      "17        COSM5029202        1011       1064del   L355Rfs*3   \n",
      "20        COSM6945747        1011  1292_1313del  F431Sfs*38   \n",
      "\n",
      "       consequence_cosmic chromosome start_genomic end_genomic  indexinternal  \n",
      "6   Deletion - Frameshift         10       8058696     8058696              5  \n",
      "8   Deletion - Frameshift         10       8073966     8073966              7  \n",
      "14  Deletion - Frameshift         10       8073909     8073916             13  \n",
      "17  Deletion - Frameshift         10       8073752     8073752             16  \n",
      "20  Deletion - Frameshift         10       8073980     8074001             18  \n",
      "   LEGACY_MUTATION_ID  id_protein      CDS     AA  \\\n",
      "9         COSM1966307        1011   471G>A  P157=   \n",
      "11        COSM6669553        1011   864C>T  C288=   \n",
      "21        COSM5569474        1011   660C>T  T220=   \n",
      "33        COSM5606574        1011   459C>T  F153=   \n",
      "35        COSM6669558        1011  1083C>T  G361=   \n",
      "\n",
      "              consequence_cosmic chromosome start_genomic end_genomic  \\\n",
      "9   Substitution - coding silent         10       8058534     8058534   \n",
      "11  Substitution - coding silent         10       8064078     8064078   \n",
      "21  Substitution - coding silent         10       8058723     8058723   \n",
      "33  Substitution - coding silent         10       8058522     8058522   \n",
      "35  Substitution - coding silent         10       8073771     8073771   \n",
      "\n",
      "    indexinternal  \n",
      "9               8  \n",
      "11             10  \n",
      "21             19  \n",
      "33             30  \n",
      "35             32  \n",
      "   LEGACY_MUTATION_ID  id_protein      CDS     AA       consequence_cosmic  \\\n",
      "10         COSM161213        1011   881T>A  M294K  Substitution - Missense   \n",
      "12        COSM8923599        1011   307C>A  L103M  Substitution - Missense   \n",
      "19        COSM1506084        1011   382G>T  G128W  Substitution - Missense   \n",
      "25        COSM6669560        1011  1199A>G  H400R  Substitution - Missense   \n",
      "26         COSM294402        1011   827G>A  R276Q  Substitution - Missense   \n",
      "\n",
      "   chromosome start_genomic end_genomic  indexinternal  \n",
      "10         10       8064095     8064095              9  \n",
      "12         10       8058370     8058370             11  \n",
      "19         10       8058445     8058445             17  \n",
      "25         10       8073887     8073887             23  \n",
      "26         10       8064041     8064041             24  \n",
      "     LEGACY_MUTATION_ID  id_protein           CDS            AA  \\\n",
      "16          COSM6904322        1011  1075_1077del       K359del   \n",
      "420         COSM6949571        1011  1070_1165del  M357_K388del   \n",
      "559         COSM6930711        1011  1036_1038del       Y346del   \n",
      "1111        COSM1238067        1011    992_994del       R331del   \n",
      "1190        COSM9521989        1011  1260_1265del  T421_P422del   \n",
      "\n",
      "       consequence_cosmic chromosome start_genomic end_genomic  indexinternal  \n",
      "16    Deletion - In frame         10       8073763     8073765             15  \n",
      "420   Deletion - In frame         10       8073758     8073853            284  \n",
      "559   Deletion - In frame         10       8069584     8069586            367  \n",
      "1111  Deletion - In frame         10       8069540     8069542            611  \n",
      "1190  Deletion - In frame         10       8073948     8073953            633  \n",
      "    LEGACY_MUTATION_ID  id_protein      CDS     AA       consequence_cosmic  \\\n",
      "94          COSM685253        1011   200C>A   S67*  Substitution - Nonsense   \n",
      "117        COSM1966369        1011   889C>T  Q297*  Substitution - Nonsense   \n",
      "169        COSM6917195        1011  1132A>T  K378*  Substitution - Nonsense   \n",
      "244        COSM6923172        1011  1172C>A  S391*  Substitution - Nonsense   \n",
      "285        COSM5752738        1011  1038C>A  Y346*  Substitution - Nonsense   \n",
      "\n",
      "    chromosome start_genomic end_genomic  indexinternal  \n",
      "94          10       8055855     8055855             82  \n",
      "117         10       8064103     8064103            100  \n",
      "169         10       8073820     8073820            134  \n",
      "244         10       8073860     8073860            183  \n",
      "285         10       8069586     8069586            211  \n",
      "    LEGACY_MUTATION_ID  id_protein               CDS          AA  \\\n",
      "110        COSM6976992        1011   992_998delinsTG  R331Mfs*20   \n",
      "150        COSM6966411        1011  947_948delinsAGT  T316Kfs*37   \n",
      "207        COSM5752118        1011       829_830insC  R277Pfs*27   \n",
      "222        COSM6943339        1011  1224_1229delinsC  P409Qfs*97   \n",
      "355        COSM9358119        1011  1092_1109delinsG       R365*   \n",
      "\n",
      "       consequence_cosmic chromosome start_genomic end_genomic  indexinternal  \n",
      "110  Complex - frameshift         10       8069540     8069546             93  \n",
      "150  Complex - frameshift         10       8069495     8069496            123  \n",
      "207  Complex - frameshift         10       8064044     8064043            157  \n",
      "222  Complex - frameshift         10       8073912     8073917            168  \n",
      "355  Complex - frameshift         10       8073780     8073797            253  \n",
      "     LEGACY_MUTATION_ID  id_protein           CDS             AA  \\\n",
      "603         COSM1163767        1011  1306_1308dup        P436dup   \n",
      "684         COSM7345115        1011  1067_1069dup        T356dup   \n",
      "744         COSM7345116        1011  1070_1075dup   M357_K358dup   \n",
      "1400        COSM5027185        1011    905_907dup        I302dup   \n",
      "1683        COSM6906117        1011    843_845dup  G281_H282insQ   \n",
      "\n",
      "        consequence_cosmic chromosome start_genomic end_genomic  indexinternal  \n",
      "603   Insertion - In frame         10       8073996     8073997            387  \n",
      "684   Insertion - In frame         10       8073757     8073758            428  \n",
      "744   Insertion - In frame         10       8073763     8073764            452  \n",
      "1400  Insertion - In frame         10       8064121     8064122            709  \n",
      "1683  Insertion - In frame         10       8064059     8064060            795  \n",
      "      LEGACY_MUTATION_ID  id_protein      CDS           AA consequence_cosmic  \\\n",
      "5548         COSM9323182        2875  2352A>T  *784Cext*14  Nonstop extension   \n",
      "7061         COSM7193446        1106  2002T>G   *668Eext*9  Nonstop extension   \n",
      "7886         COSM4749301         971   686A>G  *229Wext*13  Nonstop extension   \n",
      "13814        COSM3748900        1733   493T>G  *165Eext*48  Nonstop extension   \n",
      "18121        COSM9578989         654  1529G>C  *510Sext*43  Nonstop extension   \n",
      "\n",
      "      chromosome start_genomic end_genomic  indexinternal  \n",
      "5548           2      55636237    55636237           3737  \n",
      "7061          11     111726967   111726967           4899  \n",
      "7886          12       6237827     6237827           5575  \n",
      "13814          6      36601987    36601987           9866  \n",
      "18121          1      32285715    32285715          13176  \n",
      "       LEGACY_MUTATION_ID  id_protein                 CDS                AA  \\\n",
      "12523         COSM1717873        1767          751_759del      A251_L253del   \n",
      "252319        COSM1161201        2073          113_115del    L38_S39delinsP   \n",
      "389221        COSM9813574        3660    1389_1392delinsT           P464del   \n",
      "418062        COSM3734668         572  2232_2249delinsAAA      E746_A750del   \n",
      "425328          COSM13549         572   2235_2251delinsAG  E746_T751delinsA   \n",
      "\n",
      "                consequence_cosmic chromosome start_genomic end_genomic  \\\n",
      "12523   Complex - deletion inframe         21      34834456    34834464   \n",
      "252319  Complex - deletion inframe         17      65056479    65056481   \n",
      "389221  Complex - deletion inframe         12     132571482   132571485   \n",
      "418062  Complex - deletion inframe          7      55174769    55174786   \n",
      "425328  Complex - deletion inframe          7      55174772    55174788   \n",
      "\n",
      "        indexinternal  \n",
      "12523            9145  \n",
      "252319         188010  \n",
      "389221         288459  \n",
      "418062         309050  \n",
      "425328         310110  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       LEGACY_MUTATION_ID  id_protein   CDS     AA consequence_cosmic  \\\n",
      "146151        COSM8367386        2056  3G>T    M1?         Frameshift   \n",
      "149442         COSM333041        1635  1A>G  M1_?4         Frameshift   \n",
      "149518        COSM5679162        1635  3G>A  M1_?4         Frameshift   \n",
      "149521         COSM320833        1635  3G>T  M1_?4         Frameshift   \n",
      "208646        COSM6498374        1873  3G>A    M1?         Frameshift   \n",
      "\n",
      "       chromosome start_genomic end_genomic  indexinternal  \n",
      "146151         21      17609142    17609142         109272  \n",
      "149442          6      27873510    27873510         111804  \n",
      "149518          6      27873508    27873508         111863  \n",
      "149521          6      27873508    27873508         111866  \n",
      "208646         16      70800898    70800898         156163  \n",
      "       LEGACY_MUTATION_ID  id_protein                              CDS  \\\n",
      "307044        COSM9840441         558              849delinsACCTCCTGCC   \n",
      "421307          COSM12389         572               2321_2322insCCACGT   \n",
      "422555          COSM12388         572                2316delinsAACCCCT   \n",
      "424050          COSM13558         572       2309_2310delinsCCAGCGTGGAT   \n",
      "618927        COSM4776093         873  299_300insCCGTAGACTGGAGACTTCAAA   \n",
      "\n",
      "                        AA           consequence_cosmic chromosome  \\\n",
      "307044     P284_G285insPAP  Complex - insertion inframe         19   \n",
      "421307        H773_V774dup  Complex - insertion inframe          7   \n",
      "422555      P772_H773insTP  Complex - insertion inframe          7   \n",
      "424050        A767_V769dup  Complex - insertion inframe          7   \n",
      "618927  G99_K100insNRRLETS  Complex - insertion inframe         23   \n",
      "\n",
      "       start_genomic end_genomic  indexinternal  \n",
      "307044      39173761    39173761         228774  \n",
      "421307      55181331    55181330         309542  \n",
      "422555      55181325    55181325         309744  \n",
      "424050      55181318    55181319         309960  \n",
      "618927      48791923    48791922         449229  \n"
     ]
    }
   ],
   "source": [
    "print(mutation['consequence_cosmic'].unique().tolist())\n",
    "print(mutation[mutation['consequence_cosmic'].isnull()].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Insertion - Frameshift'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Deletion - Frameshift'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Substitution - coding silent'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Substitution - Missense'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Deletion - In frame'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Substitution - Nonsense'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Complex - frameshift'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Insertion - In frame'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Nonstop extension'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Complex - deletion inframe'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Frameshift'].head())\n",
    "print(mutation[mutation['consequence_cosmic'] == 'Complex - insertion inframe'].head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cad313a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def separar_en_cols(df, column, conseq, conseq_regex, override=False):\n",
    "    '''\n",
    "    recibe un DataFrame, el nombre de una columna auxiliar (column)\n",
    "    y un string con el tipo de consecuencia (conseq). La col. auxiliar\n",
    "    es una tupla con los elementos implicados en una mutacion\n",
    "    como la siguiente (aa1, start_pos, aa2, end_pos, aa/s_nuevos).\n",
    "    Devuelve el DataFrame df con estas 5 nuevas columnas\n",
    "    '''\n",
    "    \n",
    "    df_crop = df[df[column].str.contains(conseq_regex)].copy()\n",
    "      \n",
    "      \n",
    "    if override:\n",
    "        df_crop['aux'] = df_crop[column].str.findall(conseq_regex).str[0]\n",
    "    else:\n",
    "        df_crop['aux'] = df_crop[column].str.findall('^([A-Z\\*]+)(\\d+)_?([A-Z\\*]+)?(\\d+)?'+conseq_regex+'(.*)$').str[0]\n",
    "  \n",
    "    df_crop = df_crop[~df_crop['aux'].isnull()]\n",
    "    \n",
    "    if conseq == \"missense\":\n",
    "        df_crop['start_aa'] = df_crop['aux'].map(lambda x: str(x[1]))\n",
    "        df_crop['end_aa'] = df_crop['aux'].map(lambda x: str(x[1]))  # mismo start y end para los que no tienen un end\n",
    "        df_crop['from'] = df_crop['aux'].map(lambda x: x[0])       \n",
    "        df_crop['to'] = df_crop['aux'].map(lambda x: x[2])\n",
    "    else:    \n",
    "        # start position\n",
    "        df_crop['start_aa'] = df_crop['aux'].map(lambda x: x[1])\n",
    "        #df_crop['start_aa'] = [n[1] for n in df_crop['aux']]\n",
    "\n",
    "        # end position\n",
    "        df_crop['end_aa'] = df_crop['aux'].map(lambda x: x[3] if x[3] != '' else x[1]) # poner en el end el start si no hay end\n",
    "        #df_crop['end_aa'] = [n[3] for n in df_crop['aux']]\n",
    "        \n",
    "        # from: es el/los aa que cambian\n",
    "        df_crop['from'] = df_crop['aux'].map(lambda x: str(x[0]) + str(x[2])) # concateno si existe mas de un aa que cambia (o sea, si es un rango)\n",
    "        #df_crop['from'] = [str(n[0]) + str(n[2]) for n in df_crop['aux']]\n",
    "        \n",
    "        # to: aa/s nuevos\n",
    "        if conseq == \"nonsense\":\n",
    "            df_crop['to'] = \"*\"\n",
    "        else:\n",
    "            df_crop['to'] = df_crop['aux'].map(lambda x: x[4])\n",
    "            #df_crop['to'] = [n[4] for n in df_crop['aux']]\n",
    "\n",
    "    # consecuencia de la mutacion\n",
    "    df_crop['consequence'] = conseq\n",
    "\n",
    "    df_crop = df_crop.drop(columns=['aux'])\n",
    "\n",
    "    return df_crop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0246f498",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all mutations 1144756\n",
      "Found 264334 synonym\n",
      "Found 1606 delins\n",
      "Found 5403 deletions\n",
      "Found 884 inserions\n",
      "Found 38272 frameshift\n",
      "Found 60124 nonsense\n",
      "Found 771186 missense\n",
      "Found 1031 duplications\n",
      "mapped mutations 1142840\n",
      "Missing mutations: 1916\n",
      "                        AA      consequence_cosmic\n",
      "79              *445Lfs*63  Insertion - Frameshift\n",
      "148             *445Rfs*31   Deletion - Frameshift\n",
      "300             *445Lfs*63  Insertion - Frameshift\n",
      "371             *445Yfs*63  Insertion - Frameshift\n",
      "1337     TAMG*441delext*23     Deletion - In frame\n",
      "...                    ...                     ...\n",
      "1723332                M1?                     NaN\n",
      "1724189                M1?                     NaN\n",
      "1724253        *182Yext*26       Nonstop extension\n",
      "1724273        *104Yext*17       Nonstop extension\n",
      "1724298                M1?                     NaN\n",
      "\n",
      "[1916 rows x 2 columns]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'\\nmutations = pd.concat(tables)\\nprint(f\"Total mutations: {mutations.shape[0]}\")\\n'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# A DataFrame for each molecular consequence\n",
    "\n",
    "mutation_used = mutation.copy()\n",
    "print(f\"all mutations {mutation.shape[0]}\")\n",
    "\n",
    "synonym = separar_en_cols(mutation_used, \"AA\", \"synonym\", \"=$\")\n",
    "print(f\"Found {synonym.shape[0]} synonym\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(synonym['indexinternal'].tolist())]\n",
    "delins = separar_en_cols(mutation_used, \"AA\", \"delins\", \"delins\")\n",
    "print(f\"Found {delins.shape[0]} delins\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(delins['indexinternal'].tolist())]\n",
    "deletions = separar_en_cols(mutation_used, \"AA\", \"deletion\", \"del$\") # finish with 'del'\n",
    "print(f\"Found {deletions.shape[0]} deletions\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(deletions['indexinternal'].tolist())]\n",
    "insertions = separar_en_cols(mutation_used, \"AA\", \"insertion\", \"(?<!del)ins\") # Negative lookbehind search!\n",
    "print(f\"Found {insertions.shape[0]} inserions\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(insertions['indexinternal'].tolist())]\n",
    "frameshift = separar_en_cols(mutation_used, \"AA\", \"frameshit\", '^([A-Z])(\\d+)_?([A-Z])?(\\d+)?fs\\*(.*)$', override= True) # expressions as 'Lys1254Terfs' are nonsense, not frameshift\n",
    "print(f\"Found {frameshift.shape[0]} frameshift\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(frameshift['indexinternal'].tolist())]\n",
    "nonsense = separar_en_cols(mutation_used, \"AA\", \"nonsense\", \"\\*$\")\n",
    "print(f\"Found {nonsense.shape[0]} nonsense\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(nonsense['indexinternal'].tolist())]\n",
    "missense = separar_en_cols(mutation_used, \"AA\", \"missense\", '^([A-Z])(\\d+)(?!\\*)([A-Z])$', override=True)\n",
    "print(f\"Found {missense.shape[0]} missense\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(missense['indexinternal'].tolist())]\n",
    "duplications = separar_en_cols(mutation_used, \"AA\", \"duplication\", \"dup\")\n",
    "print(f\"Found {duplications.shape[0]} duplications\")\n",
    "\n",
    "#faltan las nonstop \\*\\d+[A-Za-z]ext*\\d+\n",
    "#los frameshift M1?, M1_?\\d*\n",
    "## Concatenate everything\n",
    "\n",
    "\n",
    "idexall = synonym['indexinternal'].tolist() + deletions['indexinternal'].tolist() + delins['indexinternal'].tolist() + duplications['indexinternal'].tolist() + frameshift['indexinternal'].tolist() + insertions['indexinternal'].tolist() + missense['indexinternal'].tolist() + nonsense['indexinternal'].tolist()\n",
    "print(f\"mapped mutations {len(idexall)}\")\n",
    "\n",
    "mutation_used = mutation_used[~mutation_used['indexinternal'].isin(duplications['indexinternal'].tolist())]\n",
    "print(f\"Missing mutations: {mutation_used.shape[0]}\")\n",
    "\n",
    "print(mutation_used[['AA', 'consequence_cosmic']])\n",
    "'''\n",
    "mutations = pd.concat(tables)\n",
    "print(f\"Total mutations: {mutations.shape[0]}\")\n",
    "'''\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e486245f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
